---
title: "Dynamics of aquatic and soil-derived bacteria in a temperate lake"
author: "Tajinder Singh, Nathan I. Wisnoski, and Jay T. Lennon"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    df_print: paged
  word_document: default
  pdf_document:
    fig_caption: yes
    keep_tex: yes
header-includes:
- \usepackage{array}
- \usepackage{graphics}
---
---


normal para text


# main header

## sub header

### sub-sub header

```{r setup, echo=FALSE}
knitr::opts_chunk$set(
  fig.width = 6,
  fig.align = "center",
  message = FALSE,
  warning = 'hide')
```

First, we'll load the packages we'll need for the analysis, as well as some other functions.

```{r packages, results='hide'}
# Import Required Packages

library(tidyverse)   
library(vegan)
library(viridis)
library(lubridate)
library(cluster)
library(dtwclust)
library(readxl)
library(writexl)


source("bin/mothur_tools.R")
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
```

Next, we'll set the aesthetics of the figures we will produce

```{r figure_setup}
my.cols <- RColorBrewer::brewer.pal(n = 4, name = "Greys")[3:4]

# Set theme for figures in the paper
theme_set(theme_classic() + 
  theme(axis.title = element_text(size = 16),
        axis.title.x = element_text(margin = margin(t = 15, b = 15)),
        axis.title.y = element_text(margin = margin(l = 15, r = 15)),
        axis.text = element_text(size = 14),
        axis.text.x = element_text(margin = margin(t = 5)),
        axis.text.y = element_text(margin = margin(r = 5)),
        #axis.line.x = element_line(size = 1),
        #axis.line.y = element_line(size = 1),
        axis.line.x = element_blank(),
        axis.line.y = element_blank(),
        axis.ticks.x = element_line(size = 1),
        axis.ticks.y = element_line(size= 1),
        axis.ticks.length = unit(.1, "in"),
        panel.border = element_rect(color = "black", fill = NA, size =1.5),
        legend.title = element_blank(),
        legend.text = element_text(size = 14),
        strip.text = element_text(size = 14),
        strip.background = element_blank()
        ))
```

## Import Data

Here, we read in the processed sequence files from mothur (shared and taxonomy) and a design of the sampling. We also load in the environmental data. We then remove the mock community from the dataset and ensure the the design and OTU table are aligned by row



```{r}
# Define Inputs
# Design = general design file for experiment
# shared = OTU table from mothur with sequence similarity clustering
# Taxonomy = Taxonomic information for each OTU
design <- "data/UL.design.txt"
shared <- "data/ul_resgrad.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared"
taxon  <- "data/ul_resgrad.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.0.03.cons.taxonomy"

# Import Design
design <- read.delim(design, header=T, row.names=1)

# Import Shared Files
otus <- read.otu(shared = shared, cutoff = "0.03")    # 97% Similarity

# Import Taxonomy
otu.taxonomy <- read.tax(taxonomy = taxon, format = "rdp")

#otus <- otus[str_which(rownames(otus), "RG"),]
otus <- otus[-which(rownames(otus) == "RGMockComm"),]
set.seed(12742899)

otus <- rrarefy(otus, min(rowSums(otus)))
print(length(which(otus == 0)))  #23630524
otus.relative.abundance <- decostand(otus, "total")
```


# Are the "persistent" reservoir taxa really representative? Look over time...

```{r time_series, fig.asp=1}
# Subset to just the time series sites
timeseries.otus <- otus[str_which(rownames(otus), "UL"),]

# make sure OTU table matches up with design order
timeseries.design <- read_csv("data/UL_timeseries_design.csv")
timeseries.otus <- timeseries.otus[match(timeseries.design$sample.name, 
                                         rownames(timeseries.otus)),]

# separate RNA and DNA dynamics
timeseries.otus.rna.og <- decostand(timeseries.otus[which(timeseries.design$sample.type == "RNA"),], 
                                 method = "total")
timeseries.otus.dna <- decostand(timeseries.otus[which(timeseries.design$sample.type == "DNA"),], 
                                 method = "total")
timeseries.otus.rna.og.presence <- decostand(timeseries.otus.rna.og, 
                                          "pa")
timeseries.otus.rna.og.present <- colnames(timeseries.otus.rna.og.presence[, colSums(timeseries.otus.rna.og.presence) > 0])

timeseries.otus.filtered <- timeseries.otus[, colnames(timeseries.otus) %in% timeseries.otus.rna.og.present]

timeseries.otus.rna <- decostand(timeseries.otus.filtered[which(timeseries.design$sample.type == "DNA"),], 
                                 method = "total")
timeseries.otus.dna.presence <- decostand(timeseries.otus.dna + timeseries.otus.rna.og, 
                                          "pa")
timeseries.otus.rna.presence <- decostand(timeseries.otus.rna, 
                                          "pa")
sum(timeseries.otus.rna.og.present != colnames(timeseries.otus.rna.presence))


timeseries.environment.data <- read.table("data/ul-seedbank.env.txt", sep="\t", header=TRUE)
timeseries.environment.data$date <- as.Date(parse_date_time(timeseries.environment.data$date, "m d y"))
timeseries.environment.data$doc[which(timeseries.environment.data$doc == "**")] <- NA
timeseries.environment.data$doc <- as.numeric(timeseries.environment.data$doc)
summary(timeseries.environment.data)
timeseries.design <- left_join(timeseries.design, timeseries.environment.data[,c("sample.id", "date")])
timeseries.environment.data <- timeseries.environment.data[-which(!(timeseries.environment.data$date %in% 
                                                                      timeseries.design$date)),]
```


```{r}
soil.otus <- otus[str_which(rownames(otus), "RGSoil"),] 
in.soils <- soil.otus[,which(colSums(soil.otus) > 0)]

soil.taxa <- colnames(in.soils)
nonsoil.taxa <- colnames(soil.otus[,which(colSums(soil.otus) == 0)])

length(soil.taxa) + length(nonsoil.taxa) == ncol(otus)


max.abundance.in.soils <- apply(decostand(soil.otus, "total"), MARGIN = 2, max)
common.in.soils <- names(max.abundance.in.soils[which(max.abundance.in.soils >= 0.001)])
rare.in.soils <- names(max.abundance.in.soils[which(max.abundance.in.soils < 0.001)])
length(common.in.soils) + length(rare.in.soils) == ncol(otus)


```




changing wide form to long form, for code and plot efficiency
```{r}
timeseries.otus.long<- timeseries.otus.rna |> as_tibble() |> 
  rownames_to_column( var= "sample.id" ) |> 
  mutate(sample.id= as.integer(sample.id)) |> 
  pivot_longer(names_to = "OTU", values_to = "relative.abundance", cols = -sample.id) |> 
  left_join(timeseries.environment.data, by= "sample.id" )

```

Plotting soil otus detected in transect from soil DNA and RNA
```{r}

#filtering transact data fractionally
transect.otus <- otus[str_which(rownames(otus), "RG"),] 
print(length(which(transect.otus == 0))) #3234533 

# separate RNA and DNA dynamics
transect.otus.rna <- transect.otus[str_which(rownames(transect.otus), "RGc"),] 
print(length(which(transect.otus.rna == 0))) #1498062 

transect.otus.dna <- transect.otus[str_which(rownames(transect.otus), "RGD"),] 
print(length(which(transect.otus.dna == 0))) #1491145 
 

transect.otus.dna.presence <- decostand(transect.otus.dna + transect.otus.rna, "pa") 
print(length(which(transect.otus.dna.presence == 1))) #10082
print(length(which(transect.otus.dna.presence == 0))) #1490092

transect.otus.rna.presence <- decostand(transect.otus.rna, "pa") 
print(length(which(transect.otus.rna.presence == 1))) #2112 
print(length(which(transect.otus.rna.presence == 0))) #1498062

transect.otus.dna2.presence <- decostand(transect.otus.dna, "pa") 
print(length(which(transect.otus.dna2.presence == 1))) #9029 
print(length(which(transect.otus.dna2.presence == 0))) #1491145

#transect.soil.1<- transect.otus.rna.presence[1, transect.otus.rna.presence[1,]==1, drop= FALSE]
#sample code to use in the loop


for (k in 1:18) {
  
  # Create a dynamic variable name
  a <- paste("transect.soil.otu.rna.", k, sep = "")
  
  b<- transect.otus.rna.presence[k, transect.otus.rna.presence[k, ] == 1, drop = FALSE]
  
  # Subset the data based on the dynamic variable name
  assign(a, b[, colnames(b) %in% soil.taxa])
  
}


transect.soil.otu.rna.all <- 
  unique(c(names(transect.soil.otu.rna.1),names(transect.soil.otu.rna.2),names(transect.soil.otu.rna.3),
           names(transect.soil.otu.rna.4),names(transect.soil.otu.rna.5),names(transect.soil.otu.rna.6),
           names(transect.soil.otu.rna.7),names(transect.soil.otu.rna.8),names(transect.soil.otu.rna.9),
           names(transect.soil.otu.rna.10),names(transect.soil.otu.rna.11),names(transect.soil.otu.rna.12),
           names(transect.soil.otu.rna.13),names(transect.soil.otu.rna.14),names(transect.soil.otu.rna.15),
           names(transect.soil.otu.rna.16),names(transect.soil.otu.rna.17),names(transect.soil.otu.rna.18)))


#220
for (k in 1:18) {
  a<- paste("transect.soil.otu.rna.", k, sep = "")
print(length(which(get(a) == 1)))
}



timeseries.otus.long |> filter( OTU %in% transect.soil.otu.rna.all) |>
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU ))+
  geom_line(alpha = 0.3)



for (k in 1:18) {
  
  # Create a dynamic variable name
  a <- paste("transect.soil.otu.dna.", k, sep = "")
  
  b<- transect.otus.dna.presence[k, transect.otus.dna.presence[k, ] == 1, drop = FALSE]
  
  # Subset the data based on the dynamic variable name
  assign(a, b[, colnames(b) %in% soil.taxa])
  
}

transect.soil.otu.dna.all <- 
  unique(c(names(transect.soil.otu.dna.1),names(transect.soil.otu.dna.2),names(transect.soil.otu.dna.3),
           names(transect.soil.otu.dna.4),names(transect.soil.otu.dna.5),names(transect.soil.otu.dna.6),
           names(transect.soil.otu.dna.7),names(transect.soil.otu.dna.8),names(transect.soil.otu.dna.9),
           names(transect.soil.otu.dna.10),names(transect.soil.otu.dna.11),names(transect.soil.otu.dna.12),
           names(transect.soil.otu.dna.13), names(transect.soil.otu.dna.14),names(transect.soil.otu.dna.15),
           names(transect.soil.otu.dna.16),names(transect.soil.otu.dna.17),names(transect.soil.otu.dna.18)))
#957


timeseries.otus.rna.present.atl.once <- 
  timeseries.otus.rna.presence[, colSums(timeseries.otus.rna.presence) >0]


transect.soil.otus.dna.present.atl.once.timeseries.rna <-
  timeseries.otus.rna.present.atl.once[,colnames(timeseries.otus.rna.present.atl.once)
                                       %in% transect.soil.otu.dna.all]
#466 instead of 536


```


Trying to plot the presence data in long form
```{r}

timeseries.otus.long.presence <- 
  timeseries.otus.rna.presence |> as_tibble() |> 
  rownames_to_column( var= "sample.id" ) |> 
  mutate(sample.id= as.integer(sample.id)) |> 
  pivot_longer(names_to = "OTU", values_to = "relative.abundance", cols = -sample.id) |> 
  left_join(timeseries.environment.data, by= "sample.id" )


```



Sorting from RNA transect data and their activity in the time series
```{r}

transect.soil.otus.rna.present.atl.once.timeseries.rna<- 
  as.data.frame(timeseries.otus.rna.present.atl.once[,colnames(timeseries.otus.rna.present.atl.once)
                                                     %in% transect.soil.otu.rna.all])

```


Attempting to Classify activity patterns based on metrics.
```{r}

names.transect.soil.otus.dna.present.atl.once.timeseries.rna<- 
  tibble( OTU= colnames(transect.soil.otus.dna.present.atl.once.timeseries.rna))

classified.transect.soil.otus.dna.present.atl.once.timeseries.rna<- 
  names.transect.soil.otus.dna.present.atl.once.timeseries.rna |> 
  mutate(active = if_else(colSums(transect.soil.otus.dna.present.atl.once.timeseries.rna)/124 > 0.9, "persistent",
                          if_else(colSums(transect.soil.otus.dna.present.atl.once.timeseries.rna)/124 < 0.1, 
                                  "transient", "seasonal")))

classified.transect.soil.otus.dna.present.atl.once.timeseries.rna |> count(active)

persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna<- 
  classified.transect.soil.otus.dna.present.atl.once.timeseries.rna |> 
  filter(active== "persistent")

timeseries.otus.long.presence |> 
  filter(OTU %in% persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna<- 
  classified.transect.soil.otus.dna.present.atl.once.timeseries.rna |> 
  filter(active== "seasonal")

timeseries.otus.long.presence |> 
  filter(OTU %in% seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna[21:40,]$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna<- 
  classified.transect.soil.otus.dna.present.atl.once.timeseries.rna |> 
  filter(active== "transient")

timeseries.otus.long.presence |> 
  filter(OTU %in% transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna[321:340,]$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

```




Manually reassigned metric classified OTU to correct for patterns based on sustained activity
```{r}
new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  read_excel("soil_otu_activity_pattern/new.sortedotus.rna.xlsx", sheet = "persistent")
new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  read_excel("soil_otu_activity_pattern/new.sortedotus.rna.xlsx", sheet = "seasonal")
new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  read_excel("soil_otu_activity_pattern/new.sortedotus.rna.xlsx", sheet = "transient")
new.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna <-
  bind_rows(new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna,
            new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna) |>
  bind_rows(new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna)

```



Visualizing re-sorted otus
```{r}


timeseries.otus.long.presence |> 
  filter(OTU %in% new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |> 
  ggplot(aes(x = date, y= relative.abundance, group= OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous(limits= c(-0.5,1.5))+
  theme_minimal()

timeseries.otus.long.presence |> 
  filter(OTU %in% new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous(limit = c(-0.5, 1.5))+
  theme_minimal()

timeseries.otus.long.presence |> 
  filter(OTU %in% new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna[31:60,]$OTU) |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous(limit = c(-0.5, 1.5))+
  theme_minimal()
```


Attempting to visualize the %contribution of the sorted otus to the dynamic active community
i.e if they were rare or common and what % of the active abundance did they make up
Only inspecting the RNA pool of the time series
```{r}
#persistent
soil.otus.persistent.active.abundance<- 
  aggregate(relative.abundance ~ sample.id, 
            data = filter(timeseries.otus.long, OTU %in%
                            new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU),
            FUN = sum) |> 
  merge(timeseries.environment.data[, c("sample.id", "date")], by = 'sample.id')

ggplot(data= soil.otus.persistent.active.abundance, 
       aes(x= date, y= relative.abundance * 100), 
       group= sample.id)+
  geom_line( alpha= 1, color = "#3F2BBD")+
  labs(
    x = "Weeks",
    y = "Percent Abundance"
  ) +
  theme(
    axis.title = element_text( face = "bold", size = 20),
    axis.text = element_text( size = 12)
  ) +
  scale_y_continuous( limits = c(0, 65))

ggsave(filename = "persistent_soil_taxa_percent_contriution.pdf", path = "./figures.rna")

#Seasonal                                
soil.otus.seasonally.active.abundance <- 
  aggregate(relative.abundance ~ sample.id, 
            data =filter(timeseries.otus.long, OTU %in% 
                          new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU),
            FUN = sum) |> 
  merge(timeseries.environment.data[, c("sample.id", "date")], by = 'sample.id')

ggplot(data= soil.otus.seasonally.active.abundance, 
       aes(x= date, y= relative.abundance * 100), 
       group= sample.id)+
  geom_line( alpha= 1, color = "#3A8639")+
  labs(
    x = "Weeks",
    y = "Percent Abundance"
  ) +
  theme(
    axis.title = element_text( face = "bold", size = 20),
    axis.text = element_text( size = 12)
  ) +
  scale_y_continuous( limits = c(0, 65))

ggsave(filename = "seasonal_soil_taxa_percent_contriution.pdf", path = "./figures.rna")


#transient
soil.otus.transient.active.abundance<- 
  aggregate(relative.abundance ~ sample.id, 
            data =filter(timeseries.otus.long, OTU %in%
                           new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU),
            FUN = sum) |> 
  merge(timeseries.environment.data[, c("sample.id", "date")], by = 'sample.id')

ggplot(
  data= soil.otus.transient.active.abundance, 
  aes(x= date, y= relative.abundance * 100), group= sample.id)+
  geom_line( alpha= 1, color = "#A42F2F")+
  labs(
    x = "Weeks",
    y = "Percent Abundance"
  ) +
  theme(
    axis.title = element_text( face = "bold", size = 20),
    axis.text = element_text( size = 12)
  ) +
  scale_y_continuous( limits = c(0, 65))

ggsave(filename = "transient_soil_taxa_percent_contriution.pdf", path = "./figures.rna")

#view(filter(otu.taxonomy, OTU %in% new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU))
#view(filter(otu.taxonomy, OTU %in% new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU))
#view(filter(otu.taxonomy, OTU %in% new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU))

```



Trying to figure out if these sorted otus are rare or common in aquatic community
```{r}

timeseries.otus.long |> 
  filter(OTU %in% new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  theme_minimal()


timeseries.otus.long |> 
  filter(OTU %in% new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()



timeseries.otus.long |> 
  filter(OTU %in% new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>   
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()

timeseries.otus.long |> 
  filter(OTU %in% new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna[91:120,]$OTU) |>
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()

```



Attempting to sort aquatic otus in the time series, i.e. otus not detected in soil samples
```{r}

timeseries.aquatic.otus.rna.present.atl.once <- 
  timeseries.otus.rna.present.atl.once[,!(colnames(timeseries.otus.rna.present.atl.once) %in% 
                                            transect.soil.otu.dna.all)]

names.timeseries.aquatic.otus.rna.present.atl.once<- 
  tibble( OTU= colnames(timeseries.aquatic.otus.rna.present.atl.once))

classified.timeseries.aquatic.otus.rna.present.atl.once <- 
  names.timeseries.aquatic.otus.rna.present.atl.once |> 
  mutate(active = if_else(colSums(timeseries.aquatic.otus.rna.present.atl.once)/124 > 0.9, "persistent", 
                          if_else(colSums(timeseries.aquatic.otus.rna.present.atl.once)/124 < 0.1, "transient",
                                  "seasonal")))

classified.timeseries.aquatic.otus.rna.present.atl.once |> count(active)

write_xlsx(classified.timeseries.aquatic.otus.rna.present.atl.once, 
           "rna.classified.timeseries.aquatic.otus.rna.present.atl.once.xlsx")


persistent.classified.timeseries.aquatic.otus.rna.present.atl.once <- 
  classified.timeseries.aquatic.otus.rna.present.atl.once |> 
  filter(active== "persistent")

write_xlsx(persistent.classified.timeseries.aquatic.otus.rna.present.atl.once,
           "rna.persistent.classified.timeseries.aquatic.otus.rna.present.atl.once.xlsx")

timeseries.otus.long.presence |> 
  filter(OTU %in% persistent.classified.timeseries.aquatic.otus.rna.present.atl.once$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

seasonal.classified.timeseries.aquatic.otus.rna.present.atl.once <- 
  classified.timeseries.aquatic.otus.rna.present.atl.once |> 
  filter(active== "seasonal")

write_xlsx(seasonal.classified.timeseries.aquatic.otus.rna.present.atl.once, 
           "rna.seasonal.classified.timeseries.aquatic.otus.rna.present.atl.once.xlsx")


timeseries.otus.long.presence |> 
  filter(OTU %in% seasonal.classified.timeseries.aquatic.otus.rna.present.atl.once[621:640,]$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

transient.classified.timeseries.aquatic.otus.rna.present.atl.once <- 
  classified.timeseries.aquatic.otus.rna.present.atl.once |> 
  filter(active== "transient")

write_xlsx(transient.classified.timeseries.aquatic.otus.rna.present.atl.once, 
           "rna.transient.classified.timeseries.aquatic.otus.rna.present.atl.once.xlsx")

timeseries.otus.long.presence |> 
  filter(OTU %in% transient.classified.timeseries.aquatic.otus.rna.present.atl.once[3801:3900,]$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

```



loading re-sorted aquatic otus
```{r}
new.persistent.classified.timeseries.aquatic.otus.rna.present.atl.once <-
  read_csv("aquatic_otu_activity_pattern/rna.new.persistent.classified.timeseries.aquatic.otus.rna.present.atl.once.csv")
new.seasonal.classified.timeseries.aquatic.otus.rna.present.atl.once <- 
  read_csv("aquatic_otu_activity_pattern/rna.new.seasonal.classified.timeseries.aquatic.otus.rna.present.atl.once.csv")
new.transient.classified.timeseries.aquatic.otus.rna.present.atl.once <- 
  read_csv("aquatic_otu_activity_pattern/rna.new.transient.classified.timeseries.aquatic.otus.rna.present.atl.once.csv")
```



plotting re-sorted aquatic otus
```{r}

timeseries.otus.long.presence |> 
  filter(OTU %in% new.persistent.classified.timeseries.aquatic.otus.rna.present.atl.once$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

timeseries.otus.long.presence |> 
  filter(OTU %in% new.seasonal.classified.timeseries.aquatic.otus.rna.present.atl.once[1:30,]$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

timeseries.otus.long.presence |> 
  filter(OTU %in% new.transient.classified.timeseries.aquatic.otus.rna.present.atl.once[1:30,]$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()
```



new aquatic %contributions
```{r}
new.aquatic.otus.persistent.active.abundance<- 
  aggregate(relative.abundance ~ sample.id, 
            data = filter(
              timeseries.otus.long, OTU %in% new.persistent.classified.timeseries.aquatic.otus.rna.present.atl.once$OTU),
            FUN = sum) |> 
  merge(timeseries.environment.data[, c("sample.id", "date")], by = 'sample.id')

ggplot(data= new.aquatic.otus.persistent.active.abundance, 
       aes(x= date, y= relative.abundance), 
       group= sample.id)+
  geom_line( alpha= 0.3)+
  theme_minimal()


new.aquatic.otus.seasonally.active.abundance<- 
  aggregate(relative.abundance ~ sample.id, 
            data =filter(timeseries.otus.long, 
                         OTU %in% new.seasonal.classified.timeseries.aquatic.otus.rna.present.atl.once$OTU),
            FUN = sum) |> 
  merge(timeseries.environment.data[, c("sample.id", "date")], by = 'sample.id')

ggplot(data= new.aquatic.otus.seasonally.active.abundance, 
       aes(x= date, y= relative.abundance), 
       group= sample.id)+
  geom_line( alpha= 0.3)+
  theme_minimal()


new.aquatic.otus.transient.active.abundance <- 
  aggregate(relative.abundance ~ sample.id, 
  data =filter(timeseries.otus.long, OTU %in%
                 new.transient.classified.timeseries.aquatic.otus.rna.present.atl.once$OTU), 
  FUN = sum) |> 
  merge(timeseries.environment.data[, c("sample.id", "date")], by = 'sample.id')


ggplot(data = new.aquatic.otus.transient.active.abundance, 
       aes(x= date, y= relative.abundance), 
       group= sample.id) +
  geom_line( alpha= 0.3)+
  theme_minimal()

```


new aquatic rare or common
```{r}
timeseries.otus.long |> 
  filter(OTU %in% new.persistent.classified.timeseries.aquatic.otus.rna.present.atl.once$OTU) |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  theme_minimal()


timeseries.otus.long |> 
  filter(OTU %in% new.persistent.classified.timeseries.aquatic.otus.rna.present.atl.once$OTU) |> 
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()



timeseries.otus.long |> 
  filter(OTU %in% new.seasonal.classified.timeseries.aquatic.otus.rna.present.atl.once[20:50,]$OTU) |> 
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()

timeseries.otus.long |> 
  filter(OTU %in% new.transient.classified.timeseries.aquatic.otus.rna.present.atl.once[450:500,]$OTU) |> 
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()
```




what in %contribution is coming from rare or common otus

for persistent OTU
```{r}

timeseries.otus.long<- 
  timeseries.otus.long |> 
  mutate(abundance.type= if_else(relative.abundance >= 0.001, 1, 0))

commonly.abundant.new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  timeseries.otus.long |> 
  filter(OTU %in% new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  filter(abundance.type == 1) |> 
  aggregate(relative.abundance ~ sample.id, FUN = sum) |> 
  rename(common.abundance = relative.abundance)

rarely.abundant.new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  timeseries.otus.long |> 
  filter(OTU %in% new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  filter(abundance.type == 0) |> 
  aggregate(relative.abundance ~ sample.id, FUN = sum)|> 
  rename(rare.abundance = relative.abundance)

aggregate.abundance.new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  soil.otus.persistent.active.abundance |> 
  rename(aggregate.abundance = relative.abundance) |>
  left_join(commonly.abundant.new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna, 
            by = 'sample.id') |>
  left_join(rarely.abundant.new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna, 
            by = 'sample.id') |> 
  mutate_all(~replace_na(.,0)) |>
  pivot_longer(cols = contains('abundance'), names_to = 'abundance.type', values_to = 'abundance')
  

ggplot(aggregate.abundance.new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna, 
       aes(x = date, y = abundance, color = abundance.type))+
  geom_line(alpha = 0.3)+
  facet_wrap(~ abundance.type) +
  scale_x_date(date_breaks = "1 year")+
  theme_minimal()

```

for seasonal otus

```{r}


commonly.abundant.new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  timeseries.otus.long |> 
  filter(OTU %in% new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  filter(abundance.type == 1) |> 
  aggregate(relative.abundance ~ sample.id, FUN = sum) |> 
  rename(common.abundance = relative.abundance)

rarely.abundant.new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  timeseries.otus.long |> 
  filter(OTU %in% new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  filter(abundance.type == 0) |> 
  aggregate(relative.abundance ~ sample.id, FUN = sum)|> 
  rename(rare.abundance = relative.abundance)

aggregate.abundance.new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  soil.otus.seasonally.active.abundance |> 
  rename(aggregate.abundance = relative.abundance) |>
  left_join(commonly.abundant.new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna, 
            by = 'sample.id') |>
  left_join(rarely.abundant.new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna, 
            by = 'sample.id') |> 
  mutate_all(~replace_na(.,0)) |>
  pivot_longer(cols = contains('abundance'), names_to = 'abundance.type', values_to = 'abundance')
  

ggplot(aggregate.abundance.new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna, 
       aes(x = date, y = abundance, color = abundance.type))+
  geom_line(alpha = 0.3)+
  facet_wrap(~ abundance.type) +
  scale_x_date(breaks = "6 months") +
  theme_minimal()

```



for transient otus
```{r}
commonly.abundant.new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  timeseries.otus.long |> 
  filter(OTU %in% new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  filter(abundance.type == 1) |> 
  aggregate(relative.abundance ~ sample.id, FUN = sum) |> 
  rename(common.abundance = relative.abundance)

rarely.abundant.new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  timeseries.otus.long |> 
  filter(OTU %in% new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  filter(abundance.type == 0) |> 
  aggregate(relative.abundance ~ sample.id, FUN = sum)|> 
  rename(rare.abundance = relative.abundance)

aggregate.abundance.new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  soil.otus.transient.active.abundance |> 
  rename(aggregate.abundance = relative.abundance) |>
  left_join(commonly.abundant.new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna, 
            by = 'sample.id') |>
  left_join(rarely.abundant.new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna, 
            by = 'sample.id') |> 
  mutate_all(~replace_na(.,0)) |>
  pivot_longer(cols = contains('abundance'), names_to = 'abundance.type', values_to = 'abundance')
  

ggplot(aggregate.abundance.new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna, 
       aes(x = sample.id, y = abundance, color = abundance.type))+
  geom_line(alpha = 0.3)+
  facet_wrap(~ abundance.type) +
  theme_minimal()


```



%contribution of soil otus to community, ALL TOGETHER
```{r}
total.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  timeseries.otus.long |> 
  filter(OTU %in% names.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  aggregate(relative.abundance ~ sample.id, FUN = sum)|> 
  rename(aggregate.abundance = relative.abundance)

rare.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  timeseries.otus.long |> 
  filter(OTU %in% names.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  filter(abundance.type == 0) |> 
  aggregate(relative.abundance ~ sample.id, FUN = sum)|> 
  rename(rare.abundance = relative.abundance)

common.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  timeseries.otus.long |> 
  filter(OTU %in% names.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  filter(abundance.type == 1) |> 
  aggregate(relative.abundance ~ sample.id, FUN = sum)|> 
  rename(common.abundance = relative.abundance)

aggregate.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna <- 
  total.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna |> 
  left_join(rare.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna, by = 'sample.id') |> 
  left_join(common.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna, by = 'sample.id') |> 
  mutate_all(~replace_na(.,0)) |> 
  pivot_longer(cols = contains('abundance'), names_to = 'abundance.type', values_to = 'abundance') |> 
  merge(timeseries.environment.data[, c("sample.id", "date")], by = 'sample.id')


ggplot(aggregate.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna, 
       aes(x = sample.id, y = abundance * 100, color = abundance.type)) +
  geom_line(alpha = 0.8) +
  facet_wrap(~abundance.type, labeller = labeller( abundance.type = 
                                    c(aggregate.abundance = "All Taxa",
                                      common.abundance = "Common Taxa", 
                                      rare.abundance = "Rare Taxa"))) +
  guides(color = "none") +
  labs(
    x = "Weeks", 
    y = " Percent Abundance \nfrom Active Soil Taxa" ) +
  scale_color_manual( values = c( "aggregate.abundance" = "#460E5F", #purple
                                  "common.abundance" = "#B16415", #orange
                                  "rare.abundance" = "#237B75" #teal
                                  )) +
  theme(
    plot.margin = unit(c(1, 1, 1, 1), "cm"), # Adjust margins
    aspect.ratio = 0.8, # Wider than it is tall
    strip.text = element_text(face = "bold"), # Make facet titles bold
    axis.title = element_text(face = "bold"), # Make axis titles bold
    plot.title = element_text(face = "bold") # Make plot title bold
  )
  

ggsave(filename = "figures.rna/percent_contribution_of_active_soil_taxa_in_aquatic_community.pdf", height = 10, width = 13, units = "in")
  


aggregate.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna.wide <- 
  total.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna |> 
  left_join(rare.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna, by = 'sample.id') |> 
  left_join(common.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna, by = 'sample.id') |> 
  mutate_all(~replace_na(.,0))

aggregate.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna.wide |> 
  mutate(ratio.common.to.rare = 
           aggregate.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna.wide$common.abundance / 
           aggregate.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna.wide$rare.abundance) |> 
  ggplot(aes(x = sample.id, y = ratio.common.to.rare)) +
  geom_line(alpha = 0.3)+
  theme_minimal()


aggregate.abundance.new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna.wide <- 
  soil.otus.persistent.active.abundance |> 
  left_join(rarely.abundant.new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna, 
            by = 'sample.id') |> 
  left_join(commonly.abundant.new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna, 
            by = 'sample.id') |> 
  mutate_all(~replace_na(.,0))

#Ratio of common to rare aggregate abundance
aggregate.abundance.new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna.wide |> 
  mutate(ratio.common.to.rare = 
           aggregate.abundance.new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna.wide$common.abundance / 
           aggregate.abundance.new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna.wide$rare.abundance) |> 
  ggplot(aes(x = sample.id, y = ratio.common.to.rare)) +
  geom_line(alpha = 0.3)+
  theme_minimal()

```


##*Skipping this part for now*
%contribution of aquatic otus to community, ALL TOGETHER
```{r}
total.abundance.timeseries.aquatic.otus.rna.present.atl.once <- 
  timeseries.otus.long |> 
  filter(OTU %in% names.timeseries.aquatic.otus.rna.present.atl.once$OTU) |>
  aggregate(relative.abundance ~ sample.id, FUN = sum)|> 
  rename(aggregate.abundance = relative.abundance)

rare.abundance.timeseries.aquatic.otus.rna.present.atl.once<- 
  timeseries.otus.long |> 
  filter(OTU %in% names.timeseries.aquatic.otus.rna.present.atl.once$OTU) |>
  filter(abundance.type == 0) |> 
  aggregate(relative.abundance ~ sample.id, FUN = sum)|> 
  rename(rare.abundance = relative.abundance)

common.abundance.timeseries.aquatic.otus.rna.present.atl.once<- 
  timeseries.otus.long |> 
  filter(OTU %in% names.timeseries.aquatic.otus.rna.present.atl.once$OTU) |>
  filter(abundance.type == 1) |> 
  aggregate(relative.abundance ~ sample.id, FUN = sum)|> 
  rename(common.abundance = relative.abundance)

aggregate.abundance.timeseries.aquatic.otus.rna.present.atl.once <- 
  total.abundance.timeseries.aquatic.otus.rna.present.atl.once |> 
  left_join(rare.abundance.timeseries.aquatic.otus.rna.present.atl.once, by = 'sample.id') |> 
  left_join(common.abundance.timeseries.aquatic.otus.rna.present.atl.once, by = 'sample.id') |> 
  mutate_all(~replace_na(.,0)) |> 
  pivot_longer(cols = contains('abundance'), names_to = 'abundance.type', values_to = 'abundance')

ggplot(aggregate.abundance.timeseries.aquatic.otus.rna.present.atl.once, 
       aes(x = sample.id, y = abundance, color = abundance.type)) +
  geom_line(alpha = 0.3)+
  facet_wrap(~abundance.type)+
  theme_minimal()


(total.abundance.timeseries.aquatic.otus.rna.present.atl.once$aggregate.abundance + 
    total.abundance.transect.soil.otus.dna.present.atl.once.timeseries.rna$aggregate.abundance)

timeseries.otus.long |> 
  filter(OTU %in% new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  filter(abundance.type == 1) |> 
  select(OTU) |> 
  unique()



```





Checking what % of active soil OTUs are common vs rare dynamically

```{r}
#All soil OTUs that were active atleast once (fixed number of OTUs in each sample.id)
percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna <-
  timeseries.otus.long |> 
  filter(OTU %in% names.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU)|>
  group_by(sample.id) |> 
  summarise(
    total.active = n(),
    total.common = sum(abundance.type == 1),
    total.rare = sum(abundance.type == 0),
    percent.common = (total.common / total.active)*100,
    percent.rare = (total.rare / total.active)*100,
  ) |> 
  mutate(mean.percent.common = mean(percent.common))


ggplot(percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna,
       aes(x= sample.id, y= percent.common)) +
  geom_line(alpha = 0.3) +
  theme_minimal()
  
#percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna$total.common +
#  percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna$total.rare == 536

#Active classified soil OTUs
percent.common.persistent.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna <-
  timeseries.otus.long |> 
  filter(OTU %in% new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU)|>
  group_by(sample.id) |> 
  summarise(
    total.active = n(),
    total.common = sum(abundance.type == 1),
    total.rare = sum(abundance.type == 0),
    percent.common = (total.common / total.active)*100,
    percent.rare = (total.rare / total.active)*100,
  ) |> 
  mutate(mean.percent.common = mean(percent.common))


ggplot(percent.common.persistent.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna,
       aes(x= sample.id, y= percent.common)) +
  geom_line(alpha = 0.3) +
  theme_minimal()

#Soil OTUs only when present (dynamic number of OTUs in each sample.id)
when.present.percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna <-
  timeseries.otus.long |> 
  mutate(presence = timeseries.otus.long.presence$relative.abundance)|>
  filter(OTU %in% names.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  filter(presence == 1) |> 
  group_by(sample.id) |> 
  summarise(
    total.present = n(),
    total.common = sum(abundance.type == 1),
    total.rare = sum(abundance.type == 0),
    percent.common = (total.common / total.present)*100,
    percent.rare = (total.rare / total.present)*100,
  ) |> 
  mutate(mean.percent.common = mean(percent.common))
#when.present.percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna$total.common +
#  when.present.percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna$total.rare == 
#  when.present.percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna$total.present

ggplot(when.present.percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna,
       aes(x= sample.id, y= percent.common)) +
  geom_line(alpha = 0.3) +
  theme_minimal()  

#persistent classified
when.present.percent.common.persistent.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna<-
  timeseries.otus.long |> 
  mutate(presence = timeseries.otus.long.presence$relative.abundance)|>
  filter(OTU %in% new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  filter(presence == 1) |> 
  group_by(sample.id) |> 
  summarise(
    total.present = n(),
    total.common = sum(abundance.type == 1),
    total.rare = sum(abundance.type == 0),
    percent.common = (total.common / total.present)*100,
    percent.rare = (total.rare / total.present)*100,
  ) |> 
  mutate(mean.percent.common = mean(percent.common))

when.present.percent.common.persistent.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna$active.type = "persistent"
#when.present.percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna$total.common +
#  when.present.percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna$total.rare == 
#  when.present.percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna$total.present

ggplot(when.present.percent.common.persistent.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna,
       aes(x= sample.id, y= percent.common)) +
  geom_line(alpha = 0.3) +
  theme_minimal() 


#transient classified
when.present.percent.common.transient.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna<-
  timeseries.otus.long |> 
  mutate(presence = timeseries.otus.long.presence$relative.abundance)|>
  filter(OTU %in% new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  filter(presence == 1) |> 
  group_by(sample.id) |> 
  summarise(
    total.present = n(),
    total.common = sum(abundance.type == 1),
    total.rare = sum(abundance.type == 0),
    percent.common = (total.common / total.present)*100,
    percent.rare = (total.rare / total.present)*100,
  ) |> 
  mutate(mean.percent.common = mean(percent.common))

when.present.percent.common.transient.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna$active.type = "transient"
#when.present.percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna$total.common +
#  when.present.percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna$total.rare == 
#  when.present.percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna$total.present

ggplot(when.present.percent.common.transient.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna, 
       aes(x= sample.id, y= percent.common)) +
  geom_line(alpha = 0.3) +
  theme_minimal() 


#seasonal Classified
when.present.percent.common.seasonal.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna<-
  timeseries.otus.long |> 
  mutate(presence = timeseries.otus.long.presence$relative.abundance)|>
  filter(OTU %in% new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  filter(presence == 1) |> 
  group_by(sample.id) |> 
  summarise(
    total.present = n(),
    total.common = sum(abundance.type == 1),
    total.rare = sum(abundance.type == 0),
    percent.common = (total.common / total.present)*100,
    percent.rare = (total.rare / total.present)*100,
  ) |> 
  mutate(mean.percent.common = mean(percent.common))

when.present.percent.common.seasonal.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna$active.type = "seasonal"
#when.present.percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna$total.common +
#  when.present.percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna$total.rare == 
#  when.present.percent.common.transect.soil.otus.rna.present.atl.once.timeseries.rna$total.present

ggplot(when.present.percent.common.seasonal.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna, 
       aes(x= sample.id, y= percent.common)) +
  geom_line(alpha = 0.3) +
  theme_minimal() 

when.present.percent.common.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna <-
  bind_rows(when.present.percent.common.persistent.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna, 
            when.present.percent.common.seasonal.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna)|>
  bind_rows(when.present.percent.common.transient.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna)

```



of the times present, what percent of time are they common?
```{r}
when.present.percent.of.time.common.transect.soil.otus.rna.present.atl.once.timeseries.rna <-
  timeseries.otus.long |> 
  mutate(presence = timeseries.otus.long.presence$relative.abundance)|>
  filter(OTU %in% names.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |>
  filter(presence == 1) |> 
  group_by(OTU) |> 
  summarise(
    total.present = n(),
    total.common = sum(abundance.type == 1),
    total.rare = sum(abundance.type == 0),
    percent.common = (total.common / total.present)*100,
    percent.rare = (total.rare / total.present)*100
  ) |> 
  mutate(mean.percent.common = mean(percent.common)) |> 
  left_join(new.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna, by = "OTU") |> 
  rename( active.type = active)
  

when.present.percent.of.time.common.alway.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna<-
  when.present.percent.of.time.common.transect.soil.otus.rna.present.atl.once.timeseries.rna |> 
  filter( OTU %in% new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |> 
  mutate(mean.percent.common = mean(percent.common))

when.present.percent.of.time.common.seasonal.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna<-
  when.present.percent.of.time.common.transect.soil.otus.rna.present.atl.once.timeseries.rna |> 
  filter( OTU %in% new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |> 
  mutate(mean.percent.common = mean(percent.common))

when.present.percent.of.time.common.transient.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna<-
  when.present.percent.of.time.common.transect.soil.otus.rna.present.atl.once.timeseries.rna |> 
  filter( OTU %in% new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna$OTU) |> 
  mutate(mean.percent.common = mean(percent.common))


```




Make box plots 1) %OTUs, 2)%of time

```{r}

when.present.percent.of.time.common.transect.soil.otus.rna.present.atl.once.timeseries.rna |>
  mutate(active.type = factor(active.type, levels = c("transient", "seasonal", "persistent"))) |> 
  ggplot(aes(x = active.type, y = percent.common, color = active.type)) +
  geom_boxplot() +
  guides( color = "none") +
  labs( x = "Activity Pattern", y = "Percent of time \ncommon when present") +
  theme(
    axis.title = element_text(face = "bold", size = 20),
    axis.text = element_text(size = 15),
    plot.margin = margin(20, 10),
    axis.title.y = element_text(hjust = 0.72)) +
  scale_color_manual( values = c( "transient" = "#A42F2F", #red
                                  "seasonal" = "#3A8639", #green
                                  "persistent" = "#3F2BBD" #blue
                                  ))

ggsave("soil_taxa_percent_of_times_common_when_present.pdf", path = "./figures.rna")

when.present.percent.common.classified.transect.soil.otus.rna.present.atl.once.timeseries.rna |> 
  mutate(active.type = factor(active.type, levels = c( "transient", "seasonal", "persistent"))) |> 
  ggplot(aes(x = active.type, y = percent.common, color = active.type)) +
  geom_boxplot() +
  guides(color = "none") +
  labs( x = "Activity Pattern", y = "Percent of taxa \ncommon when present") +
  theme(
    axis.title = element_text(face = "bold", size = 20),
    axis.text = element_text(size = 15),
    plot.margin = margin(20, 20),
    axis.title.y = element_text( hjust = 0.72)) +
  scale_color_manual( values = c( "transient" = "#A42F2F", #red
                                  "seasonal" = "#3A8639", #green
                                  "persistent" = "#3F2BBD" #blue
                                  ))

ggsave("percent_of_soil_taxa_common_when_present.pdf", path = "./figures.rna")

```


Getting example OTUs to represent different activity patterns

```{r}
#persistent: selected Otu00001
timeseries.otus.long.presence |> 
  filter(OTU == "Otu00001") |> 
  ggplot(aes(x = date, y = relative.abundance)) +
  facet_wrap(~OTU) +
  geom_line( alpha = 1, color = "#3F2BBD") +
  scale_y_continuous( limits = c(-0.5, 1.5)) +
  labs(
    x = "Weeks",
    y = ""
  ) +
  theme(
    axis.title = element_text( face = "bold", size = 20),
    axis.text.x = element_text( size = 12)
  )

ggsave("persistent_example_otu.pdf", path = "./figures.rna")


#Seasonal: selected Otu00078
timeseries.otus.long.presence |> 
  filter(OTU == "Otu00078") |> 
  ggplot(aes(x = date, y = relative.abundance)) +
  facet_wrap(~OTU) +
  geom_line( alpha = 1, color = "#3A8639") +
  scale_y_continuous( limits = c(-0.5, 1.5)) +
  labs(
    x = "Weeks",
    y = ""
  ) +
  theme(
    axis.title = element_text( face = "bold", size = 20),
    axis.text.x = element_text( size = 12)
  )

ggsave("seasonal_example_otu.pdf", path = "./figures.rna")



#transient: selected Otu00042
timeseries.otus.long.presence |> 
  filter(OTU == "Otu00042") |> 
  ggplot(aes(x = date, y = relative.abundance)) +
  facet_wrap(~OTU,) +
  geom_line( alpha = 1, color = "#A42F2F" ) +
  scale_y_continuous( limits = c(-0.5, 1.5)) +
  labs(
    x = "Weeks",
    y = ""
  ) +
  theme(
    axis.title = element_text( face = "bold", size = 20),
    axis.text.x = element_text( size = 12),
  )

ggsave("transient_example_otu.pdf", path = "./figures.rna")


```





##*need to do this part*
shifting to calling terrestrial-derived taxa as shared taxa. 
Identifying all the shared taxa in the active aquatic pool and assign rare/common in soils
Assigning rare/common relative abundance for transect taxa
Figures plotting max. relative abundance of shared taxa in soil and transect and their temporal pattern

```{r}

soil.taxa.relative.abundance <- 
  in.soils |> 
  as.data.frame() |> 
  decostand("total") |> 
  rownames_to_column(var = "sampleid") |> 
  pivot_longer(names_to = "OTU", values_to = "rel.abundance.soil" , cols = -sampleid) |>
  mutate(abundance.type.soil = if_else( rel.abundance.soil > 0.001, 1, 0))

soil.taxa.maximum.relative.abundance <- 
  soil.taxa.relative.abundance |> 
  group_by(OTU) |> 
  mutate( max.rel.abundance.soil = max(rel.abundance.soil)) |>
  ungroup() |> 
  mutate(max.abundance.type.soil = if_else( max.rel.abundance.soil > 0.001, 1, 0))

transect.otu.dna.relative.abundance <- 
  transect.otus.dna |>
  as.data.frame() |> 
  decostand("total") |>
  t() |> as.data.frame() |> 
  filter(rowSums(pick(everything())) > 0)

transect.otu.dna.max.relative.abundance <-
  transect.otu.dna.relative.abundance |> 
  mutate( max.rel.abundance.transect = apply(pick(everything()), 1, max)) |> 
  mutate(max.abundance.type.transect = if_else( max.rel.abundance.transect > 0.001, 1, 0)) |> 
  rownames_to_column( var = "OTU") |> 
  select(OTU, 
         max.rel.abundance.transect,
         max.abundance.type.transect)
  
transect.otu.rna.relative.abundance <- 
  transect.otus.rna |>
  as.data.frame() |> 
  decostand("total") |>
  t() |> as.data.frame() |> 
  filter(rowSums(pick(everything())) > 0)

transect.otu.rna.max.relative.abundance <-
  transect.otu.rna.relative.abundance |> 
  mutate( rna.max.rel.abundance.transect = apply(pick(everything()), 1, max)) |> 
  mutate(rna.max.abundance.type.transect = if_else( rna.max.rel.abundance.transect > 0.001, 1, 0)) |> 
  rownames_to_column( var = "OTU") |> 
  select(OTU, 
         rna.max.rel.abundance.transect,
         rna.max.abundance.type.transect)

names.shared.otus.timeseries.rna.in.soils <- 
  timeseries.otus.rna.present.atl.once |>
  as.data.frame() |> 
  select(any_of(colnames(in.soils))) |> 
  colnames()


combined.new.classified.otus.rna.present.atl.once <- 
  bind_rows(new.persistent.classified.timeseries.aquatic.otus.rna.present.atl.once,
          new.persistent.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna, 
          new.seasonal.classified.timeseries.aquatic.otus.rna.present.atl.once,
          new.seasonal.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna,
          new.transient.classified.timeseries.aquatic.otus.rna.present.atl.once,
          new.transient.classified.transect.soil.otus.dna.present.atl.once.timeseries.rna) |>
  rename(timeseries.active.type = "active")


shared.otus.timeseries.rna.in.soils <-
  data.frame(OTU = names.shared.otus.timeseries.rna.in.soils) |> 
  left_join(soil.taxa.maximum.relative.abundance, by = "OTU") |> 
  left_join(transect.otu.dna.max.relative.abundance, by = "OTU") |> 
  left_join(combined.new.classified.otus.rna.present.atl.once, by = "OTU") |>
  left_join(transect.otu.rna.max.relative.abundance, by = "OTU") 




shared.otus.timeseries.rna.in.soils |> 
  ggplot(aes(x = max.rel.abundance.soil, y = max.rel.abundance.transect, 
             color = timeseries.active.type))+
  geom_hline( yintercept = 0.001, color = "grey") +
  geom_vline( xintercept = 0.001, color = "grey") +
  geom_point() +
  labs(x = "Max. Rel. Abundance\n in Soil", y = "Max. Rel. Abundance\n in Aquatic Transect") +
  annotate("text", x = 10^(-1), y = 10^(-1), label = "DNA", size = 6, adj = 1) +
  scale_x_log10(limits = c(0.0001, 0.1), labels = scales::label_log()) +
  scale_y_log10(limits = c(0.0001, 0.1), labels = scales::label_log()) +
  theme(legend.position = "none") +
  coord_fixed( ratio = 1) +
  theme(legend.position = c(0.98,0.7), 
        legend.justification = c(1, 0), 
        legend.box.margin = margin(t = 0, r = 0, b = 0, l = 0),
        legend.text = element_text(face = "bold"),
        legend.background = element_rect(linewidth = 0.3, colour = "black"),
         strip.text = element_text(face = "bold", size = 14), # Make facet titles bold
        axis.title = element_text(face = "bold", size = 14), # Make axis titles bold
        plot.title = element_text(face = "bold", size = 14)) # Make plot title bold)

ggsave(filename = "figures.rna/newfigure.dna.pdf", width = 7, height = 6, units = "in")

shared.otus.timeseries.rna.in.soils |> 
  ggplot(aes(x = max.rel.abundance.soil, y = rna.max.rel.abundance.transect, 
             color =timeseries.active.type))+
  geom_hline( yintercept = 0.001, color = "grey") +
  geom_vline( xintercept = 0.001, color = "grey") +
  geom_point() +
  labs(x = "Max. Rel. Abundance \n in Soil", y = "Max Rel. Abundance \n in Aquatic Transect") +
  annotate("text", x = 10^(-1), y = 10^(-1), label = "RNA", size = 6, adj = 1) +
  scale_x_log10(limits = c(0.0001, 0.1), labels = scales::label_log()) +
  scale_y_log10(limits = c(0.0001, 0.1), labels = scales::label_log()) +
  coord_fixed( ratio = 1) +
  theme(legend.position = c(0.98,0.7), 
        legend.justification = c(1, 0), 
        legend.box.margin = margin(t = 0, r = 0, b = 0, l = 0),
        legend.text = element_text(face = "bold"),
        legend.background = element_rect(linewidth = 0.3, colour = "black"),
        strip.text = element_text(face = "bold", size = 14), # Make facet titles bold
        axis.title = element_text(face = "bold", size = 14), # Make axis titles bold
        plot.title = element_text(face = "bold", size = 14)) # Make plot title bold)

ggsave(filename = "figures.rna/newfigure.rna.pdf", width = 7, height = 6, units = "in")


  
```
