---
title: "Dynamics of aquatic and soil-derived bacteria in a temperate lake"
author: "Tajinder Singh, Nathan I. Wisnoski, and Jay T. Lennon"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    df_print: paged
  word_document: default
  pdf_document:
    fig_caption: yes
    keep_tex: yes
header-includes:
- \usepackage{array}
- \usepackage{graphics}
---
---


normal para text


# main header

## sub header

### sub-sub header

```{r setup, echo=FALSE}
knitr::opts_chunk$set(
  fig.width = 6,
  fig.align = "center",
  message = FALSE,
  warning = 'hide')
```

First, we'll load the packages we'll need for the analysis, as well as some other functions.

```{r packages, results='hide'}
# Import Required Packages

library(tidyverse)   
library(vegan)
library(viridis)
library(lubridate)

source("bin/mothur_tools.R")
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
```

Next, we'll set the aesthetics of the figures we will produce

```{r figure_setup}
my.cols <- RColorBrewer::brewer.pal(n = 4, name = "Greys")[3:4]

# Set theme for figures in the paper
theme_set(theme_classic() + 
  theme(axis.title = element_text(size = 16),
        axis.title.x = element_text(margin = margin(t = 15, b = 15)),
        axis.title.y = element_text(margin = margin(l = 15, r = 15)),
        axis.text = element_text(size = 14),
        axis.text.x = element_text(margin = margin(t = 5)),
        axis.text.y = element_text(margin = margin(r = 5)),
        #axis.line.x = element_line(size = 1),
        #axis.line.y = element_line(size = 1),
        axis.line.x = element_blank(),
        axis.line.y = element_blank(),
        axis.ticks.x = element_line(size = 1),
        axis.ticks.y = element_line(size= 1),
        axis.ticks.length = unit(.1, "in"),
        panel.border = element_rect(color = "black", fill = NA, size =1.5),
        legend.title = element_blank(),
        legend.text = element_text(size = 14),
        strip.text = element_text(size = 14),
        strip.background = element_blank()
        ))
```

## Import Data

Here, we read in the processed sequence files from mothur (shared and taxonomy) and a design of the sampling. We also load in the environmental data. We then remove the mock community from the dataset and ensure the the design and OTU table are aligned by row



```{r}
# Define Inputs
# Design = general design file for experiment
# shared = OTU table from mothur with sequence similarity clustering
# Taxonomy = Taxonomic information for each OTU
design <- "data/UL.design.txt"
shared <- "data/ul_resgrad.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared"
taxon  <- "data/ul_resgrad.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.0.03.cons.taxonomy"

# Import Design
design <- read.delim(design, header=T, row.names=1)

# Import Shared Files
OTUs <- read.otu(shared = shared, cutoff = "0.03")    # 97% Similarity

# Import Taxonomy
OTU.tax <- read.tax(taxonomy = taxon, format = "rdp")

#OTUs <- OTUs[str_which(rownames(OTUs), "RG"),]
OTUs <- OTUs[-which(rownames(OTUs) == "RGMockComm"),]

OTUs <- rrarefy(OTUs, min(rowSums(OTUs)))
OTUs.REL <- decostand(OTUs, "total")
```


# Are the "persistent" reservoir taxa really representative? Look over time...

```{r time_series, fig.asp=1}
# Subset to just the time series sites
UL.ts.OTUs <- OTUs[str_which(rownames(OTUs), "UL"),]

# make sure OTU table matches up with design order
UL.ts.design <- read_csv("data/UL_timeseries_design.csv")
UL.ts.OTUs <- UL.ts.OTUs[match(UL.ts.design$sample.name, rownames(UL.ts.OTUs)),]

# separate RNA and DNA dynamics
UL.ts.OTUs.RNA <- decostand(UL.ts.OTUs[which(UL.ts.design$sample.type == "RNA"),], method = "total")
UL.ts.OTUs.DNA <- decostand(UL.ts.OTUs[which(UL.ts.design$sample.type == "DNA"),], method = "total")

UL.ts.OTUs.DNA.PA <- decostand(UL.ts.OTUs.DNA + UL.ts.OTUs.RNA, "pa")
UL.ts.OTUs.RNA.PA <- decostand(UL.ts.OTUs.RNA, "pa")

env.ts.data <- read.table("data/ul-seedbank.env.txt", sep="\t", header=TRUE)
env.ts.data$date <- as.Date(parse_date_time(env.ts.data$date, "m d y"))
env.ts.data$doc[which(env.ts.data$doc == "**")] <- NA
env.ts.data$doc <- as.numeric(env.ts.data$doc)
summary(env.ts.data)
UL.ts.design <- left_join(UL.ts.design, env.ts.data[,c("sample.id", "date")])
env.ts.data <- env.ts.data[-which(!(env.ts.data$date %in% UL.ts.design$date)),]
```


```{r}
RGSoil.OTUs <- OTUs[str_which(rownames(OTUs), "RGSoil"),] 
in.soils <- RGSoil.OTUs[,which(colSums(RGSoil.OTUs) > 0)]

soil.taxa <- colnames(in.soils)
nonsoil.taxa <- colnames(RGSoil.OTUs[,which(colSums(RGSoil.OTUs) == 0)])

length(soil.taxa) + length(nonsoil.taxa) == ncol(OTUs)

max.abund.in.soils <- apply(decostand(RGSoil.OTUs, "total"), MARGIN = 2, max)
common.in.soils <- names(max.abund.in.soils[which(max.abund.in.soils >= 0.001)])
rare.in.soils <- names(max.abund.in.soils[which(max.abund.in.soils < 0.001)])
length(common.in.soils) + length(rare.in.soils) == ncol(OTUs)

# pull out time series of taxa in soils
UL.ts.toplot.RNA <- UL.ts.OTUs.RNA[,which(colMeans(UL.ts.OTUs.DNA.PA) > .2)] # present in half samples

OTUs.toplot.in.soil <- UL.ts.OTUs.RNA[, which(colnames(UL.ts.toplot.RNA) %in% soil.taxa)]

# pull out taxa not detected in soils
OTUs.toplot.not.in.soil <- UL.ts.OTUs.RNA[, which(colnames(UL.ts.toplot.RNA) %in% nonsoil.taxa)]

ncol(OTUs.toplot.in.soil) + ncol(OTUs.toplot.not.in.soil) == ncol(UL.ts.toplot.RNA)

otu_trends_to_plot <- cbind.data.frame(UL.ts.design[which(UL.ts.design$sample.type == "RNA"),], OTUs.toplot.in.soil) %>% as_tibble() %>% 
  gather(-sample.name, -sample.type, -sample.id, -date, key = OTU, value = rel_abund) %>% 
  mutate(soils = ifelse(OTU %in% nonsoil.taxa, "Absent from soils",
                   ifelse(OTU %in% common.in.soils, "Common in soils", "Rare in soils")))
otu_trends_to_plot %>% 
  ggplot(aes(x = date, y = rel_abund + 1e-4, group = OTU)) +
  geom_point(alpha = .1) + 
  geom_line(color = "blue",
            alpha = 0.5) +
  geom_vline(aes(xintercept = as_date("2013-07-15"))) +
  scale_y_log10() + 
  scale_x_date(labels = scales::date_format(format = "%b %y")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  facet_wrap(~soils, ncol = 1) +
  labs(x = "",
       y = "Relative abundance")
  

```



Many of them do appear to track the seasons quite well, suggesting there could be a seasonality component to the role of terrestrial inputs into the reservoir.


```{r}
otu_trends_to_plot %>% 
  group_by(OTU) %>% 
  filter(mean(rel_abund) > 1e-3) %>% 
  ggplot(aes(x = date, y = rel_abund + 1e-4, group = OTU, color = soils)) +
  geom_point(alpha = .1) + 
  geom_line(color = "blue",
            alpha = 0.5) +
  geom_line(stat = "smooth", se = FALSE, span = 0.5) +
  geom_vline(aes(xintercept = as_date("2013-07-15"))) +
  scale_y_log10() + 
  scale_x_date(labels = scales::date_format(format = "%b %y")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  facet_wrap(~OTU) +
  labs(x = "",
       y = "Relative abundance")

```



Proportion of Soil OTUs richness in the Seed bank


```{r}
SB.ts.OTUs.PA <- UL.ts.OTUs.DNA.PA-UL.ts.OTUs.RNA.PA
head(SB.ts.OTUs.PA[,1:10])

#create vector for seed bank richness

SB.ts.richness <- rowSums(SB.ts.OTUs.PA)

plot(SB.ts.richness, type = "l")

#obtaining soil OTU richness in SB

SB.ts.Soil.OTUs.PA <- subset(SB.ts.OTUs.PA,select = colnames(in.soils) )
SB.ts.Soil.richness <- rowSums(SB.ts.Soil.OTUs.PA)
identical(SB.ts.Soil.richness, SB.ts.Soil.richness)


png("figures/sbsoilrichness.png", width = 8, height = 6, res= 500, units = "in")
plot(SB.ts.Soil.richness/SB.ts.richness, type= "l")
dev.off(which = 2)
```


Making a smooth line graph

```{r}
#making a data frame for proportion of soil OTUs in SB

SB.ts.Soil.Prop <- tibble(sample.id= env.ts.data$sample.id, date= env.ts.data$date, Soil.Prop= SB.ts.Soil.richness/SB.ts.richness)

#plotting proportion over time

sb.soil.richness.sm <- ggplot(SB.ts.Soil.Prop, aes(x= date, y= Soil.Prop))+
  geom_point()+
  geom_smooth()
ggsave(filename = "figures/sbsoilprop.pdf", plot = sb.soil.richness.sm, height = 6, width = 8) 

```



changing wide form to long form
```{r}

OTU.ts.long<- UL.ts.OTUs.RNA |> as_tibble() |> 
  rownames_to_column( var= "sample.id" ) |> 
  mutate(sample.id= as.integer(sample.id)) |> 
  pivot_longer(names_to = "OTU", values_to = "relative.abundance", cols = -sample.id) |> 
  left_join(env.ts.data, by= "sample.id" )

OTU.ts.long |> filter(OTU %in% RG.avg.ac) |> 
  filter( OTU %in% soil.taxa) |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU))+
  geom_line(alpha = 0.3)+
  scale_y_log10()


```



Classifying Middle OTUs in RG samples

```{r}

RG.middle.OTU <-OTUs.REL[c("RGD08", "RGD09", "RGD10", "RGD11", "RGc08", "RGc09", "RGc10", "RGc11"),]
RG.middle.OTU <-OTUs.REL[c("RGD08", "RGD09", "RGD10", "RGD11", "RGc08", "RGc09", "RGc10", "RGc11"),]

#OTUs always active
RG.middle.RNA <- as.data.frame(t(colMeans(RG.middle.OTU[str_which(rownames(RG.middle.OTU), "RGc"), ])))
RG.middle.Ac <- RG.middle.RNA[, which(RG.middle.RNA[1, ] > 0)]
RG.avg.ac<- colnames(RG.middle.Ac)
RG.avg.soil.ac <- RG.middle.Ac[, which(colnames(RG.middle.Ac) %in% soil.taxa)]

#OTUs never active
RG.middle.DNA <- as.data.frame(t(colMeans(RG.middle.OTU[str_which(rownames(RG.middle.OTU), "RGD"),])))
RG.middle.sb <- RG.middle.DNA |> 
  select(where(~. >0)) |> 
  select(!matches(RG.avg.ac))
RG.avg.sb <- colnames(RG.middle.sb)
RG.avg.soil.sb <- RG.middle.sb[, which(colnames(RG.middle.sb) %in% soil.taxa)]

 
```



Classifying middle OTUs based on relative abundance
```{r}


RG.clsfd.ac <- RG.middle.Ac |>
  t() |>
  as_tibble() |> 
  rename(rel.ab = V1) |> 
  mutate(ab.class = if_else(rel.ab< exp(-10), 6, 
                            if_else(rel.ab < exp(-9), 5, 
                            if_else(rel.ab < exp(-8), 4, 
                                    if_else(rel.ab < exp(-6), 3, 
                                            if_else(rel.ab < exp(-4), 2, 
                                                    if_else(rel.ab < exp(-2), 1, 12)))))))


row.names(RG.clsfd.ac) <- names(RG.middle.Ac)
RG.clsfd.ac<- rownames_to_column(RG.clsfd.ac, var = "OTU")

RG.clsfd.sb<- RG.middle.sb |>
  t() |> 
  as_tibble() |> 
  rename(rel.ab = V1) |> 
  mutate(ab.class = if_else(rel.ab< exp(-10), 6, 
                            if_else(rel.ab < exp(-9), 5, 
                            if_else(rel.ab < exp(-8), 4, 
                                    if_else(rel.ab < exp(-6), 3, 
                                            if_else(rel.ab < exp(-4), 2, 
                                                    if_else(rel.ab < exp(-2), 1, 12)))))))

row.names(RG.clsfd.sb) <- names(RG.middle.sb)
RG.clsfd.sb<- rownames_to_column(RG.clsfd.sb, var = "OTU")
```






Graphing Soil OTUs based on classification
```{r}

OTU.ts.long |> filter(OTU %in% RG.avg.ac) |> 
  filter(OTU %in% soil.taxa) |>
  left_join(RG.clsfd.ac, by = "OTU") |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU))+
  geom_line(alpha = 0.3)+
  scale_y_log10()+
  facet_wrap(~ab.class)



OTU.ts.long |> filter(OTU %in% RG.avg.ac) |> 
  filter(!OTU %in% soil.taxa) |>
  left_join(RG.clsfd.ac, by = "OTU") |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU))+
  geom_line(alpha = 0.3)+
  scale_y_log10()+
  facet_wrap(~ab.class)




#getting taxonomy of different rel. abundance classes
clsfd.ac.1<- filter(RG.clsfd.ac, ab.class == 1)
so.ac.1<- OTU.tax |>  
  filter( OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.ac.1$OTU)

aq.ac.1<- OTU.tax |> 
  filter(!OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.ac.1$OTU)

clsfd.ac.2<- filter(RG.clsfd.ac, ab.class == 2)
so.ac.2<- OTU.tax |> 
  filter( OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.ac.2$OTU)

aq.ac.2<- OTU.tax |>  
  filter(!OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.ac.2$OTU)

clsfd.ac.3<- filter(RG.clsfd.ac, ab.class == 3)
so.ac.3<- OTU.tax |>  
  filter( OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.ac.3$OTU)

aq.ac.3<- OTU.tax |>  
  filter(!OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.ac.3$OTU)

clsfd.ac.4<- filter(RG.clsfd.ac, ab.class == 4)
so.ac.4<- OTU.tax |>  
  filter( OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.ac.4$OTU)

aq.ac.4<- OTU.tax |> 
  filter(!OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.ac.4$OTU)

clsfd.ac.5<- filter(RG.clsfd.ac, ab.class == 5)
so.ac.5<- OTU.tax |> 
  filter( OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.ac.5$OTU)

aq.ac.5<- OTU.tax |> 
  filter(!OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.ac.5$OTU)

clsfd.ac.6<- filter(RG.clsfd.ac, ab.class == 6)
so.ac.6<- OTU.tax |> 
  filter( OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.ac.6$OTU)

aq.ac.6<- OTU.tax |> 
  filter(!OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.ac.6$OTU)
  
```




```{r}
OTU.ts.long |> filter(OTU %in% RG.avg.sb) |> 
  filter(OTU %in% soil.taxa) |>
  left_join(RG.clsfd.sb, by = "OTU") |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU))+
  geom_line(alpha = 0.3)+
  scale_y_log10()+
  facet_wrap(~ab.class)

OTU.ts.long |> filter(OTU %in% RG.avg.sb) |> 
  filter(!OTU %in% soil.taxa) |>
  left_join(RG.clsfd.sb, by = "OTU") |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU))+
  geom_line(alpha = 0.3)+
  scale_y_log10()+
  facet_wrap(~ab.class)

#getting taxonomy of different rel. abundance classes

clsfd.sb.1<- filter(RG.clsfd.sb, ab.class == 1)
so.sb.1<- OTU.tax |> 
  filter( OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.sb.1$OTU)

aq.sb.1<- OTU.tax |> 
  filter(OTU %in% clsfd.sb.1$OTU) |> 
  filter(!OTU %in% soil.taxa)

clsfd.sb.2<- filter(RG.clsfd.sb, ab.class == 2)
so.sb.2<- OTU.tax |>
  filter( OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.sb.2$OTU)

aq.sb.2<- OTU.tax |> 
  filter(OTU %in% clsfd.sb.2$OTU) |> 
  filter(!OTU %in% soil.taxa)

clsfd.sb.3<- filter(RG.clsfd.sb, ab.class == 3)
so.sb.3<- OTU.tax |>
  filter( OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.sb.3$OTU)

aq.sb.3<- OTU.tax |> 
  filter(OTU %in% clsfd.sb.3$OTU) |> 
  filter(!OTU %in% soil.taxa)

clsfd.sb.4<- filter(RG.clsfd.sb, ab.class == 4)
so.sb.4<- OTU.tax |>
  filter( OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.sb.4$OTU)

aq.sb.4<- OTU.tax |> 
  filter(OTU %in% clsfd.sb.4$OTU) |> 
  filter(!OTU %in% soil.taxa)

clsfd.sb.5<- filter(RG.clsfd.sb, ab.class == 5)
so.sb.5<- OTU.tax |> 
  filter( OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.sb.5$OTU)

aq.sb.5<- OTU.tax |> 
  filter(OTU %in% clsfd.sb.5$OTU) |> 
  filter(!OTU %in% soil.taxa)

clsfd.sb.6<- filter(RG.clsfd.sb, ab.class == 6)
so.sb.6<- OTU.tax |> 
  filter( OTU %in% soil.taxa) |>
  filter(OTU %in% clsfd.sb.6$OTU)

aq.sb.6<- OTU.tax |> 
  filter(OTU %in% clsfd.sb.6$OTU) |> 
  filter(!OTU %in% soil.taxa)

  

```

 

  


 
  


