---
title: "Dynamics of aquatic and soil-derived bacteria in a temperate lake"
author: "Tajinder Singh, Nathan I. Wisnoski, and Jay T. Lennon"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    df_print: paged
  word_document: default
  pdf_document:
    fig_caption: yes
    keep_tex: yes
header-includes:
- \usepackage{array}
- \usepackage{graphics}
---
---


normal para text


# main header

## sub header

### sub-sub header

```{r setup, echo=FALSE}
knitr::opts_chunk$set(
  fig.width = 6,
  fig.align = "center",
  message = FALSE,
  warning = 'hide')
```

First, we'll load the packages we'll need for the analysis, as well as some other functions.

```{r packages, results='hide'}
# Import Required Packages

library(tidyverse)   
library(vegan)
library(viridis)
library(lubridate)
library(cluster)
library(dtwclust)
library(readxl)
library(writexl)


source("bin/mothur_tools.R")
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
```

Next, we'll set the aesthetics of the figures we will produce

```{r figure_setup}
my.cols <- RColorBrewer::brewer.pal(n = 4, name = "Greys")[3:4]

# Set theme for figures in the paper
theme_set(theme_classic() + 
  theme(axis.title = element_text(size = 16),
        axis.title.x = element_text(margin = margin(t = 15, b = 15)),
        axis.title.y = element_text(margin = margin(l = 15, r = 15)),
        axis.text = element_text(size = 14),
        axis.text.x = element_text(margin = margin(t = 5)),
        axis.text.y = element_text(margin = margin(r = 5)),
        #axis.line.x = element_line(size = 1),
        #axis.line.y = element_line(size = 1),
        axis.line.x = element_blank(),
        axis.line.y = element_blank(),
        axis.ticks.x = element_line(size = 1),
        axis.ticks.y = element_line(size= 1),
        axis.ticks.length = unit(.1, "in"),
        panel.border = element_rect(color = "black", fill = NA, size =1.5),
        legend.title = element_blank(),
        legend.text = element_text(size = 14),
        strip.text = element_text(size = 14),
        strip.background = element_blank()
        ))
```

## Import Data

Here, we read in the processed sequence files from mothur (shared and taxonomy) and a design of the sampling. We also load in the environmental data. We then remove the mock community from the dataset and ensure the the design and OTU table are aligned by row



```{r}
# Define Inputs
# Design = general design file for experiment
# shared = OTU table from mothur with sequence similarity clustering
# Taxonomy = Taxonomic information for each OTU
design <- "data/UL.design.txt"
shared <- "data/ul_resgrad.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared"
taxon  <- "data/ul_resgrad.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.0.03.cons.taxonomy"

# Import Design
design <- read.delim(design, header=T, row.names=1)

# Import Shared Files
OTUs <- read.otu(shared = shared, cutoff = "0.03")    # 97% Similarity

# Import Taxonomy
OTU.tax <- read.tax(taxonomy = taxon, format = "rdp")

#OTUs <- OTUs[str_which(rownames(OTUs), "RG"),]
OTUs <- OTUs[-which(rownames(OTUs) == "RGMockComm"),]
set.seed(12742899)

OTUs <- rrarefy(OTUs, min(rowSums(OTUs)))
print(length(which(OTUs == 0)))  #23630524
OTUs.REL <- decostand(OTUs, "total")
```


# Are the "persistent" reservoir taxa really representative? Look over time...

```{r time_series, fig.asp=1}
# Subset to just the time series sites
UL.ts.OTUs <- OTUs[str_which(rownames(OTUs), "UL"),]

# make sure OTU table matches up with design order
UL.ts.design <- read_csv("data/UL_timeseries_design.csv")
UL.ts.OTUs <- UL.ts.OTUs[match(UL.ts.design$sample.name, rownames(UL.ts.OTUs)),]

# separate RNA and DNA dynamics
UL.ts.OTUs.RNA <- decostand(UL.ts.OTUs[which(UL.ts.design$sample.type == "RNA"),], method = "total")
UL.ts.OTUs.DNA <- decostand(UL.ts.OTUs[which(UL.ts.design$sample.type == "DNA"),], method = "total")

UL.ts.OTUs.DNA.PA <- decostand(UL.ts.OTUs.DNA + UL.ts.OTUs.RNA, "pa")
UL.ts.OTUs.RNA.PA <- decostand(UL.ts.OTUs.RNA, "pa")

env.ts.data <- read.table("data/ul-seedbank.env.txt", sep="\t", header=TRUE)
env.ts.data$date <- as.Date(parse_date_time(env.ts.data$date, "m d y"))
env.ts.data$doc[which(env.ts.data$doc == "**")] <- NA
env.ts.data$doc <- as.numeric(env.ts.data$doc)
summary(env.ts.data)
UL.ts.design <- left_join(UL.ts.design, env.ts.data[,c("sample.id", "date")])
env.ts.data <- env.ts.data[-which(!(env.ts.data$date %in% UL.ts.design$date)),]
```


```{r}
RGSoil.OTUs <- OTUs[str_which(rownames(OTUs), "RGSoil"),] 
in.soils <- RGSoil.OTUs[,which(colSums(RGSoil.OTUs) > 0)]

soil.taxa <- colnames(in.soils)
nonsoil.taxa <- colnames(RGSoil.OTUs[,which(colSums(RGSoil.OTUs) == 0)])

length(soil.taxa) + length(nonsoil.taxa) == ncol(OTUs)


max.abund.in.soils <- apply(decostand(RGSoil.OTUs, "total"), MARGIN = 2, max)
common.in.soils <- names(max.abund.in.soils[which(max.abund.in.soils >= 0.001)])
rare.in.soils <- names(max.abund.in.soils[which(max.abund.in.soils < 0.001)])
length(common.in.soils) + length(rare.in.soils) == ncol(OTUs)

# pull out time series of taxa in soils
UL.ts.toplot.RNA <- UL.ts.OTUs.RNA[,which(colMeans(UL.ts.OTUs.DNA.PA) > .2)] # present in half samples

OTUs.toplot.in.soil <- UL.ts.OTUs.RNA[, which(colnames(UL.ts.toplot.RNA) %in% soil.taxa)]

# pull out taxa not detected in soils
OTUs.toplot.not.in.soil <- UL.ts.OTUs.RNA[, which(colnames(UL.ts.toplot.RNA) %in% nonsoil.taxa)]

ncol(OTUs.toplot.in.soil) + ncol(OTUs.toplot.not.in.soil) == ncol(UL.ts.toplot.RNA)

otu_trends_to_plot <- cbind.data.frame(UL.ts.design[which(UL.ts.design$sample.type == "RNA"),], OTUs.toplot.in.soil) %>% as_tibble() %>% 
  gather(-sample.name, -sample.type, -sample.id, -date, key = OTU, value = rel_abund) %>% 
  mutate(soils = ifelse(OTU %in% nonsoil.taxa, "Absent from soils",
                   ifelse(OTU %in% common.in.soils, "Common in soils", "Rare in soils")))
otu_trends_to_plot %>% 
  ggplot(aes(x = date, y = rel_abund + 1e-4, group = OTU)) +
  geom_point(alpha = .1) + 
  geom_line(color = "blue",
            alpha = 0.5) +
  geom_vline(aes(xintercept = as_date("2013-07-15"))) +
  scale_y_log10() + 
  scale_x_date(labels = scales::date_format(format = "%b %y")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  facet_wrap(~soils, ncol = 1) +
  labs(x = "",
       y = "Relative abundance")
  

```







Classifying Middle OTUs in RG samples, the classification will help focus on a smaller
section of the OTUs closer to the Time series collection to get a better
activity representation instead of far way points which might be too different.

```{r}

RG.middle.OTU <-OTUs.REL[c("RGD08", "RGD09", "RGD10", "RGD11", "RGc08", "RGc09", "RGc10", "RGc11"),]


#OTUs always active
RG.middle.RNA <- as.data.frame(t(colMeans(RG.middle.OTU[str_which(rownames(RG.middle.OTU), "RGc"), ])))
RG.middle.Ac <- RG.middle.RNA[, which(RG.middle.RNA[1, ] > 0)]
RG.avg.ac<- colnames(RG.middle.Ac)
RG.avg.soil.ac <- RG.middle.Ac[, which(colnames(RG.middle.Ac) %in% soil.taxa)]

#OTUs never active
RG.middle.DNA <- as.data.frame(t(colMeans(RG.middle.OTU[str_which(rownames(RG.middle.OTU), "RGD"),])))

 
```



changing wide form to long form, for code and plot efficiency
```{r}
OTU.ts.long<- UL.ts.OTUs.RNA |> as_tibble() |> 
  rownames_to_column( var= "sample.id" ) |> 
  mutate(sample.id= as.integer(sample.id)) |> 
  pivot_longer(names_to = "OTU", values_to = "relative.abundance", cols = -sample.id) |> 
  left_join(env.ts.data, by= "sample.id" )

OTU.ts.long |> filter(OTU %in% RG.avg.ac) |> 
  filter( OTU %in% soil.taxa) |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU))+
  geom_line(alpha = 0.3)+
  scale_y_log10()

```

Plotting soil OTUs detected in transect from soil DNA and RNA
```{r}

#filtering transact data fractionally
RG.OTUs <- OTUs[str_which(rownames(OTUs), "RG"),] 
print(length(which(RG.OTUs == 0))) #3234533 

# separate RNA and DNA dynamics
RG.OTUs.RNA <- RG.OTUs[str_which(rownames(RG.OTUs), "RGc"),] 
print(length(which(RG.OTUs.RNA == 0))) #1498062 

RG.OTUs.DNA <- RG.OTUs[str_which(rownames(RG.OTUs), "RGD"),] 
print(length(which(RG.OTUs.DNA == 0))) #1491145 
 

RG.OTUs.DNA.PA <- decostand(RG.OTUs.DNA + RG.OTUs.RNA, "pa") 
print(length(which(RG.OTUs.DNA.PA == 1))) #10082
print(length(which(RG.OTUs.DNA.PA == 0))) #1490092

RG.OTUs.RNA.PA <- decostand(RG.OTUs.RNA, "pa") 
print(length(which(RG.OTUs.RNA.PA == 1))) #2112 
print(length(which(RG.OTUs.RNA.PA == 0))) #1498062

RG.OTUs.DNA2.PA <- decostand(RG.OTUs.DNA, "pa") 
print(length(which(RG.OTUs.DNA2.PA == 1))) #9029 
print(length(which(RG.OTUs.DNA2.PA == 0))) #1491145

#RG.1<- RG.OTUs.RNA.PA[1, RG.OTUs.RNA.PA[1,]==1, drop= FALSE]
#sample code to use in the loop


for (k in 1:18) {
  
  # Create a dynamic variable name
  a <- paste("RG.", k, sep = "")
  
  b<- RG.OTUs.RNA.PA[k, RG.OTUs.RNA.PA[k, ] == 1, drop = FALSE]
  
  # Subset the data based on the dynamic variable name
  assign(a, b[, colnames(b) %in% soil.taxa])
  
}


RG.all <- unique(c(names(RG.1),names(RG.2),names(RG.3),names(RG.4),names(RG.5),names(RG.6),names(RG.7),
  names(RG.8),names(RG.9),names(RG.10),names(RG.11),names(RG.12),names(RG.13),names(RG.14),names(RG.15),
            names(RG.16),names(RG.17),names(RG.18)))
#220
for (k in 1:18) {
  a<- paste("RG.", k, sep = "")
print(length(which(get(a) == 1)))
}



OTU.ts.long |> filter( OTU %in% RG.all) |>
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU ))+
  geom_line(alpha = 0.3)



for (k in 1:18) {
  
  # Create a dynamic variable name
  a <- paste("RG.D", k, sep = "")
  
  b<- RG.OTUs.DNA.PA[k, RG.OTUs.DNA.PA[k, ] == 1, drop = FALSE]
  
  # Subset the data based on the dynamic variable name
  assign(a, b[, colnames(b) %in% soil.taxa])
  
}

RG.D.all <- unique(c(names(RG.D1),names(RG.D2),names(RG.D3),names(RG.D4),names(RG.D5),
                      names(RG.D6),
                     names(RG.D7),names(RG.D8),names(RG.D9),names(RG.D10),names(RG.D11),
                     names(RG.D12),names(RG.D13), names(RG.D14),names(RG.D15),
            names(RG.D16),names(RG.D17),names(RG.D18)))
#957

for (k in 1:18) {
  a<- paste("RG.D", k, sep = "")
print(length(which(get(a) == 1)))
}


OTU.ts.long |> filter( OTU %in% RG.D.all) |>
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU ))+
  geom_line(alpha = 0.3)




UL.OTUs.RNA.AP<- UL.ts.OTUs.RNA.PA[, colSums(UL.ts.OTUs.RNA.PA) >0]


UL.OTUs.RNA.APS<- UL.OTUs.RNA.AP[,colnames(UL.OTUs.RNA.AP) %in% RG.D.all]
#536


OTU.ts.long |> filter( OTU %in% colnames(UL.OTUs.RNA.APS)) |>
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU ))+
  geom_line(alpha = 0.3)+
  scale_y_log10()



OTU.ts.long |> filter( OTU %in% colnames(UL.OTUs.RNA.APS[,31:60])) |>
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU ))+
  geom_line(alpha = 0.3)+
  facet_wrap(~ OTU, scales = "free_y")


```
Studying Sorted Soil Soil OTUs in Time series
(these were manually sorted)
Loading objects
```{r}
ul.dna.soilotus.always <- read_excel("soil_otu_activity_pattern/sortedotus.xlsx", sheet = "ul.dna.soilotus.always")

ul.dna.soilotus.blips <- read_excel("soil_otu_activity_pattern/sortedotus.xlsx", sheet = "ul.dna.soilotus.blips")

ul.dna.soilotus.fh <- read_excel("soil_otu_activity_pattern/sortedotus.xlsx", sheet = "ul.dna.soilotus.fh")

ul.dna.soilotus.s1 <- read_excel("soil_otu_activity_pattern/sortedotus.xlsx", sheet = "ul.dna.soilotus.s1")

ul.dna.soilotus.s2 <- read_excel("soil_otu_activity_pattern/sortedotus.xlsx", sheet = "ul.dna.soilotus.s2")

ul.dna.soilotus.s3 <- read_excel("soil_otu_activity_pattern/sortedotus.xlsx", sheet = "ul.dna.soilotus.s3")

```



Looking at Dynamics
```{r}
OTU.ts.long |> filter( OTU %in% ul.dna.soilotus.always$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU ))+
  geom_line(alpha = 0.3)+
  scale_y_log10()

OTU.ts.long |> filter( OTU %in% ul.dna.soilotus.blips$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU ))+
  geom_line(alpha = 0.3)+
  scale_y_log10()

OTU.ts.long |> filter( OTU %in% (ul.dna.soilotus.always[1:50,])$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU ))+
  geom_line(alpha = 0.3)+
  facet_wrap(~ OTU, scales = "free_y")

```



Trying to plot the PA data in long form
```{r}
OTU.ts.long.pa<- UL.ts.OTUs.RNA.PA |> as_tibble() |> 
  rownames_to_column( var= "sample.id" ) |> 
  mutate(sample.id= as.integer(sample.id)) |> 
  pivot_longer(names_to = "OTU", values_to = "relative.abundance", cols = -sample.id) |> 
  left_join(env.ts.data, by= "sample.id" )

OTU.ts.long.pa |> filter( OTU %in% ul.dna.soilotus.always$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU ))+
  geom_line(alpha = 0.3)+
  #scale_y_log10()+
  facet_wrap(~OTU)

OTU.ts.long.pa |> filter( OTU %in% ul.dna.soilotus.fh$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU ))+
  geom_line(alpha = 0.3)+
  scale_y_log10()+
  facet_wrap(~OTU)

OTU.ts.long.pa |> filter( OTU %in% (ul.dna.soilotus.blips[1:50,])$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU ))+
  geom_line(alpha = 0.3)+
  facet_wrap(~ OTU, scales = "free_y")


```



Sorting from RNA transect data and their activity in the time series
```{r}

UL.OTUs.RNA.APS.R<- as.data.frame(UL.OTUs.RNA.AP[,colnames(UL.OTUs.RNA.AP) %in% RG.all])

OTU.ts.long |> filter( OTU %in% colnames(UL.OTUs.RNA.APS.R)) |>
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU ))+
  geom_line(alpha = 0.3)+
  scale_y_log10()

OTU.ts.long |> filter( OTU %in% colnames(UL.OTUs.RNA.APS.R[,1:30])) |>
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU ))+
  geom_line(alpha = 0.3)+
  facet_wrap(~ OTU, scales = "free_y")

```


Attempting to Classify activity patterns based on metrics.
```{r}
RG.D.APS<- tibble( OTU= colnames(UL.OTUs.RNA.APS))

sorted.otu.clfd<- RG.D.APS |> 
  mutate(active = if_else(colSums(UL.OTUs.RNA.APS)/124 > 0.9, "always", 
                          if_else(colSums(UL.OTUs.RNA.APS)/124 < 0.1, "blips", "seasonal")))
sorted.otu.clfd |> count(active)

sorted.otu.always<- sorted.otu.clfd |> 
  filter(active== "always")

OTU.ts.long.pa |> filter(OTU %in% sorted.otu.always$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

sorted.otu.seasonal<- sorted.otu.clfd |> 
  filter(active== "seasonal")

OTU.ts.long.pa |> filter(OTU %in% sorted.otu.seasonal[1:35,]$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

sorted.otu.blips<- sorted.otu.clfd |> 
  filter(active== "blips")

OTU.ts.long.pa |> filter(OTU %in% sorted.otu.blips[1:35,]$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

```




Manually reassigned metric classified OTU to correct for patterns based on sustained activity
```{r}
new.sorted.otu.always <- read_excel("soil_otu_activity_pattern/new.sortedotus.xlsx", sheet = "always")
new.sorted.otu.seasonal <- read_excel("soil_otu_activity_pattern/new.sortedotus.xlsx", sheet = "seasonal")
new.sorted.otu.blips <- read_excel("soil_otu_activity_pattern/new.sortedotus.xlsx", sheet = "blips")

```



Visualizing re-sorted OTUs
```{r}


OTU.ts.long.pa |> filter(OTU %in% new.sorted.otu.always$OTU) |> 
  ggplot(aes(x = sample.id, y= relative.abundance, group= OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous(limits= c(-0.5,1.5))+
  theme_minimal()

OTU.ts.long.pa |> filter(OTU %in% new.sorted.otu.seasonal$OTU) |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous(limit = c(-0.5, 1.5))+
  theme_minimal()

OTU.ts.long.pa |> filter(OTU %in% new.sorted.otu.blips[31:60,]$OTU) |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous(limit = c(-0.5, 1.5))+
  theme_minimal()
```



Attempting to visualize the %contribution of the sorted OTUs to the dynamic active community
i.e if they were rare or common and what % of the active abundance did they make up
Only inspecting the RNA pool of the time series
```{r}

always.abundance<- aggregate(relative.abundance ~ sample.id, 
                                   data = filter(OTU.ts.long, OTU %in% new.sorted.otu.always$OTU),
                                   FUN = sum)
ggplot(data= always.abundance, aes(x= sample.id, y= relative.abundance), group= sample.id)+
  geom_line( alpha= 0.3)+
  theme_minimal()


seasonal.abundance<- aggregate(relative.abundance ~ sample.id, 
                                   data =filter(OTU.ts.long, OTU %in% new.sorted.otu.seasonal$OTU),
                                   FUN = sum)
ggplot(data= seasonal.abundance, aes(x= sample.id, y= relative.abundance), group= sample.id)+
  geom_line( alpha= 0.3)+
  theme_minimal()


blips.abundance<- aggregate(relative.abundance ~ sample.id, 
                                   data =filter(OTU.ts.long, OTU %in% new.sorted.otu.blips$OTU),
                                   FUN = sum)
ggplot(data= blips.abundance, aes(x= sample.id, y= relative.abundance), group= sample.id)+
  geom_line( alpha= 0.3)+
  theme_minimal()


#view(filter(OTU.tax, OTU %in% new.sorted.otu.always$OTU))
#view(filter(OTU.tax, OTU %in% new.sorted.otu.seasonal$OTU))
#view(filter(OTU.tax, OTU %in% new.sorted.otu.blips$OTU))

```



Trying to figure out if these sorted OTUs are rare or common in aquatic community
```{r}

OTU.ts.long |> 
  filter(OTU %in% new.sorted.otu.always$OTU) |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  theme_minimal()


OTU.ts.long |> 
  filter(OTU %in% new.sorted.otu.always$OTU) |> 
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()



OTU.ts.long |> 
  filter(OTU %in% new.sorted.otu.seasonal$OTU) |> 
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()

OTU.ts.long |> 
  filter(OTU %in% new.sorted.otu.blips[91:120,]$OTU) |> 
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()
```






Attempting to sort aquatic OTUs in the time series, i.e. OTUs not detected in soil samples
```{r}

UL.OTUs.RNA.APA<- UL.OTUs.RNA.AP[,!(colnames(UL.OTUs.RNA.AP) %in% RG.D.all)]

UL.OTUs.APA<- tibble( OTU= colnames(UL.OTUs.RNA.APA))

sorted.otu.clfd.a<- UL.OTUs.APA |> 
  mutate(active = if_else(colSums(UL.OTUs.RNA.APA)/124 > 0.9, "always", 
                          if_else(colSums(UL.OTUs.RNA.APA)/124 < 0.1, "blips", "seasonal")))
sorted.otu.clfd.a |> count(active)
write_xlsx(sorted.otu.clfd.a, "sorted.otu.clfd.a.xlsx")
sorted.otu.always.a<- sorted.otu.clfd.a |> 
  filter(active== "always")
write_xlsx(sorted.otu.always.a, "sorted.otu.always.a.xlsx")

OTU.ts.long.pa |> filter(OTU %in% sorted.otu.always.a$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

sorted.otu.seasonal.a<- sorted.otu.clfd.a |> 
  filter(active== "seasonal")
write_xlsx(sorted.otu.seasonal.a, "sorted.otu.seasonal.a.xlsx")

OTU.ts.long.pa |> filter(OTU %in% sorted.otu.seasonal.a[661:690,]$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

sorted.otu.blips.a<- sorted.otu.clfd.a |> 
  filter(active== "blips")
write_xlsx(sorted.otu.blips.a, "sorted.otu.blips.a.xlsx")

OTU.ts.long.pa |> filter(OTU %in% sorted.otu.blips.a[101:200,]$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

```



%contribution
```{r}
always.abundance.a<- aggregate(relative.abundance ~ sample.id, 
                                   data = filter(OTU.ts.long, OTU %in% sorted.otu.always.a$OTU),
                                   FUN = sum)
ggplot(data= always.abundance.a, aes(x= sample.id, y= relative.abundance), group= sample.id)+
  geom_line( alpha= 0.3)+
  theme_minimal()


seasonal.abundance.a<- aggregate(relative.abundance ~ sample.id, 
                                   data =filter(OTU.ts.long, OTU %in% sorted.otu.seasonal.a$OTU),
                                   FUN = sum)
ggplot(data= seasonal.abundance.a, aes(x= sample.id, y= relative.abundance), group= sample.id)+
  geom_line( alpha= 0.3)+
  theme_minimal()


blips.abundance.a<- aggregate(relative.abundance ~ sample.id, 
                                   data =filter(OTU.ts.long, OTU %in% sorted.otu.blips.a$OTU),
                                   FUN = sum)
ggplot(data= blips.abundance.a, aes(x= sample.id, y= relative.abundance), group= sample.id)+
  geom_line( alpha= 0.3)+
  theme_minimal()



#view(filter(OTU.tax, OTU %in% new.sorted.otu.always$OTU))
#view(filter(OTU.tax, OTU %in% new.sorted.otu.seasonal$OTU))
#view(filter(OTU.tax, OTU %in% new.sorted.otu.blips$OTU))
```


aquatic OG rare or common
```{r}
OTU.ts.long |> 
  filter(OTU %in% sorted.otu.always.a$OTU) |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  theme_minimal()


OTU.ts.long |> 
  filter(OTU %in% sorted.otu.always.a$OTU) |> 
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()



OTU.ts.long |> 
  filter(OTU %in% sorted.otu.seasonal.a[20:50,]$OTU) |> 
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()

OTU.ts.long |> 
  filter(OTU %in% sorted.otu.blips.a[450:500,]$OTU) |> 
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()
```


loading re-sorted aquatic otus
```{r}
new.always.a <- read_csv("aquatic_otu_activity_pattern/new.always.a.csv")
new.seasonal.a <- read_csv("aquatic_otu_activity_pattern/new.seasonal.a.csv")
new.blips.a <- read_csv("aquatic_otu_activity_pattern/new.blips.a.csv")
```



plotting re-sorted aquatic otus
```{r}

OTU.ts.long.pa |> filter(OTU %in% new.always.a$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

OTU.ts.long.pa |> filter(OTU %in% new.seasonal.a[1:30,]$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()

OTU.ts.long.pa |> filter(OTU %in% new.blips.a[1:30,]$OTU) |>
  ggplot(aes(x= sample.id, y= relative.abundance,group = OTU))+
  geom_line(alpha = 0.3)+
  facet_wrap(~OTU)+
  scale_y_continuous( limits = c(-0.5,1.5))+
  theme_minimal()
```



new aquatic %contributions
```{r}
new.always.abundance.a<- aggregate(relative.abundance ~ sample.id, 
                                   data = filter(OTU.ts.long, OTU %in% new.always.a$OTU),
                                   FUN = sum)
ggplot(data= new.always.abundance.a, aes(x= sample.id, y= relative.abundance), group= sample.id)+
  geom_line( alpha= 0.3)+
  theme_minimal()


new.seasonal.abundance.a<- aggregate(relative.abundance ~ sample.id, 
                                   data =filter(OTU.ts.long, OTU %in% new.seasonal.a$OTU),
                                   FUN = sum)
ggplot(data= new.seasonal.abundance.a, aes(x= sample.id, y= relative.abundance), group= sample.id)+
  geom_line( alpha= 0.3)+
  theme_minimal()


new.blips.abundance.a<- aggregate(relative.abundance ~ sample.id, 
                                   data =filter(OTU.ts.long, OTU %in% new.blips.a$OTU),
                                   FUN = sum)
ggplot(data= new.blips.abundance.a, aes(x= sample.id, y= relative.abundance), group= sample.id)+
  geom_line( alpha= 0.3)+
  theme_minimal()

```


new aquatic rare or common
```{r}
OTU.ts.long |> 
  filter(OTU %in% new.always.a$OTU) |> 
  ggplot(aes(x= sample.id, y= relative.abundance, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  theme_minimal()


OTU.ts.long |> 
  filter(OTU %in% new.always.a$OTU) |> 
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()



OTU.ts.long |> 
  filter(OTU %in% new.seasonal.a[20:50,]$OTU) |> 
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()

OTU.ts.long |> 
  filter(OTU %in% new.blips.a[450:500,]$OTU) |> 
  mutate(type= if_else(relative.abundance >= 0.001, 1, 0)) |> 
  ggplot(aes(x= sample.id, y= type, group= OTU))+
  geom_line( alpha =0.3)+
  facet_wrap(~ OTU)+
  scale_y_continuous(limits = c(-0.5, 1.5))+
  theme_minimal()
```




what in %contribution is coming from rare or common OTUs

for always OTU
```{r}

OTU.ts.long<- OTU.ts.long |> 
  mutate(abundance.type= if_else(relative.abundance >= 0.001, 1, 0))

always.common.abundance <- OTU.ts.long |> 
  filter(OTU %in% new.sorted.otu.always$OTU) |>
  filter(abundance.type == 1) |> 
  aggregate(relative.abundance ~ sample.id, 
                             FUN = sum) |> 
  rename(common.abundance = relative.abundance)

always.rare.abundance <- OTU.ts.long |> 
  filter(OTU %in% new.sorted.otu.always$OTU) |>
  filter(abundance.type == 0) |> 
  aggregate(relative.abundance ~ sample.id, 
            FUN = sum)|> 
  rename(rare.abundance = relative.abundance)

always.abundance.soil.otus <- always.abundance |> 
  rename(aggregate.abundance = relative.abundance) |>
  left_join(always.common.abundance, by = 'sample.id') |>
  left_join(always.rare.abundance, by = 'sample.id') |> 
  mutate_all(~replace_na(.,0)) |>
  pivot_longer(cols = contains('abundance'), names_to = 'abundance.type', values_to = 'abundance')
  

ggplot(always.abundance.soil.otus, aes(x = sample.id, y = abundance, color = abundance.type))+
  geom_line(alpha = 0.3)+
  facet_wrap(~ abundance.type) +
  theme_minimal()

```

for seasonal OTUs

```{r}


seasonal.common.abundance <- OTU.ts.long |> 
  filter(OTU %in% new.sorted.otu.seasonal$OTU) |>
  filter(abundance.type == 1) |> 
  aggregate(relative.abundance ~ sample.id, 
                             FUN = sum) |> 
  rename(common.abundance = relative.abundance)

seasonal.rare.abundance <- OTU.ts.long |> 
  filter(OTU %in% new.sorted.otu.seasonal$OTU) |>
  filter(abundance.type == 0) |> 
  aggregate(relative.abundance ~ sample.id, 
            FUN = sum)|> 
  rename(rare.abundance = relative.abundance)

seasonal.abundance.soil.otus <- seasonal.abundance |> 
  rename(aggregate.abundance = relative.abundance) |>
  left_join(seasonal.common.abundance, by = 'sample.id') |>
  left_join(seasonal.rare.abundance, by = 'sample.id') |> 
  mutate_all(~replace_na(.,0)) |>
  pivot_longer(cols = contains('abundance'), names_to = 'abundance.type', values_to = 'abundance')
  

ggplot(seasonal.abundance.soil.otus, aes(x = sample.id, y = abundance, color = abundance.type))+
  geom_line(alpha = 0.3)+
  facet_wrap(~ abundance.type) +
  theme_minimal()
```



for blips OTUs
```{r}
blips.common.abundance <- OTU.ts.long |> 
  filter(OTU %in% new.sorted.otu.blips$OTU) |>
  filter(abundance.type == 1) |> 
  aggregate(relative.abundance ~ sample.id, 
                             FUN = sum) |> 
  rename(common.abundance = relative.abundance)

blips.rare.abundance <- OTU.ts.long |> 
  filter(OTU %in% new.sorted.otu.blips$OTU) |>
  filter(abundance.type == 0) |> 
  aggregate(relative.abundance ~ sample.id, 
            FUN = sum)|> 
  rename(rare.abundance = relative.abundance)

blips.abundance.soil.otus <- blips.abundance |> 
  rename(aggregate.abundance = relative.abundance) |>
  left_join(blips.common.abundance, by = 'sample.id') |>
  left_join(blips.rare.abundance, by = 'sample.id') |> 
  mutate_all(~replace_na(.,0)) |>
  pivot_longer(cols = contains('abundance'), names_to = 'abundance.type', values_to = 'abundance')
  

ggplot(blips.abundance.soil.otus, aes(x = sample.id, y = abundance, color = abundance.type))+
  geom_line(alpha = 0.3)+
  facet_wrap(~ abundance.type) +
  theme_minimal()
```



%contribution of soil OTUs to community, ALL TOGETHER
```{r}
aggregate.soil.otus <- OTU.ts.long |> 
  filter(OTU %in% RG.D.APS$OTU) |>
  aggregate(relative.abundance ~ sample.id, 
            FUN = sum)|> 
  rename(aggregate.abundance = relative.abundance)

rare.abundance.soil.otus<- OTU.ts.long |> 
  filter(OTU %in% RG.D.APS$OTU) |>
  filter(abundance.type == 0) |> 
  aggregate(relative.abundance ~ sample.id, 
            FUN = sum)|> 
  rename(rare.abundance = relative.abundance)

common.abundance.soil.otus<- OTU.ts.long |> 
  filter(OTU %in% RG.D.APS$OTU) |>
  filter(abundance.type == 1) |> 
  aggregate(relative.abundance ~ sample.id, 
            FUN = sum)|> 
  rename(common.abundance = relative.abundance)

abundance.soil.otus <- aggregate.soil.otus |> 
  left_join(rare.abundance.soil.otus, by = 'sample.id') |> 
  left_join(common.abundance.soil.otus, by = 'sample.id') |> 
  mutate_all(~replace_na(.,0)) |> 
  pivot_longer(cols = contains('abundance'), names_to = 'abundance.type', values_to = 'abundance')

ggplot(abundance.soil.otus, aes(x = sample.id, y = abundance, color = abundance.type)) +
  geom_line(alpha = 0.3)+
  facet_wrap(~abundance.type)+
  theme_minimal()
abundance.soil.otus.wide <- aggregate.soil.otus |> 
  left_join(rare.abundance.soil.otus, by = 'sample.id') |> 
  left_join(common.abundance.soil.otus, by = 'sample.id') |> 
  mutate_all(~replace_na(.,0))

abundance.soil.otus.wide |> 
  mutate(ratio.common.to.rare = abundance.soil.otus.wide$common.abundance/abundance.soil.otus.wide$rare.abundance) |> 
  ggplot(aes(x = sample.id, y = ratio.common.to.rare)) +
  geom_line(alpha = 0.3)+
  theme_minimal()


always.abundance.soil.otus.wide <- always.abundance |> 
  left_join(always.rare.abundance, by = 'sample.id') |> 
  left_join(always.common.abundance, by = 'sample.id') |> 
  mutate_all(~replace_na(.,0))

a<- always.abundance.soil.otus.wide |> 
  mutate(ratio.common.to.rare = always.abundance.soil.otus.wide$common.abundance/always.abundance.soil.otus.wide$rare.abundance) |> 
  ggplot(aes(x = sample.id, y = ratio.common.to.rare)) +
  geom_line(alpha = 0.3)+
  theme_minimal()
```



%contribution of aquatic OTUs to community, ALL TOGETHER
```{r}
aggregate.aquatic.otus <- OTU.ts.long |> 
  filter(OTU %in% UL.OTUs.APA$OTU) |>
  aggregate(relative.abundance ~ sample.id, 
            FUN = sum)|> 
  rename(aggregate.abundance = relative.abundance)

rare.abundance.aquatic.otus<- OTU.ts.long |> 
  filter(OTU %in% UL.OTUs.APA$OTU) |>
  filter(abundance.type == 0) |> 
  aggregate(relative.abundance ~ sample.id, 
            FUN = sum)|> 
  rename(rare.abundance = relative.abundance)

common.abundance.aquatic.otus<- OTU.ts.long |> 
  filter(OTU %in% UL.OTUs.APA$OTU) |>
  filter(abundance.type == 1) |> 
  aggregate(relative.abundance ~ sample.id, 
            FUN = sum)|> 
  rename(common.abundance = relative.abundance)

abundance.aquatic.otus <- aggregate.aquatic.otus |> 
  left_join(rare.abundance.aquatic.otus, by = 'sample.id') |> 
  left_join(common.abundance.aquatic.otus, by = 'sample.id') |> 
  mutate_all(~replace_na(.,0)) |> 
  pivot_longer(cols = contains('abundance'), names_to = 'abundance.type', values_to = 'abundance')

ggplot(abundance.aquatic.otus, aes(x = sample.id, y = abundance, color = abundance.type)) +
  geom_line(alpha = 0.3)+
  facet_wrap(~abundance.type)+
  theme_minimal()


(aggregate.aquatic.otus$aggregate.abundance + aggregate.soil.otus$aggregate.abundance)

OTU.ts.long |> 
  filter(OTU %in% new.sorted.otu.blips$OTU) |>
  filter(abundance.type == 1) |> 
  select(OTU) |> 
  unique()



```

